{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "ea4444bb-b96c-49c7-89c7-ce3b5eeaf278",
   "metadata": {},
   "source": [
    "# Install Required Software"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "dd27a9d8-86ba-4566-aaba-c1397831637a",
   "metadata": {
    "scrolled": true,
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\n",
      "                                 Dload  Upload   Total   Spent    Left  Speed\n",
      "  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0\n",
      "  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0\n",
      "100 90.4M  100 90.4M    0     0  94.5M      0 --:--:-- --:--:-- --:--:--  210M\n",
      "PREFIX=/home/jupyter/mambaforge\n",
      "Unpacking payload ...\n",
      "Extracting python-3.10.6-ha86cf86_0_cpython.tar.bz2\n",
      "Extracting _libgcc_mutex-0.1-conda_forge.tar.bz2\n",
      "Extracting ca-certificates-2022.9.24-ha878542_0.tar.bz2\n",
      "Extracting ld_impl_linux-64-2.39-hc81fddc_0.tar.bz2\n",
      "Extracting libstdcxx-ng-12.2.0-h46fd767_19.tar.bz2\n",
      "Extracting pybind11-abi-4-hd8ed1ab_3.tar.bz2\n",
      "Extracting tzdata-2022f-h191b570_0.tar.bz2\n",
      "Extracting libgomp-12.2.0-h65d4601_19.tar.bz2\n",
      "Extracting _openmp_mutex-4.5-2_gnu.tar.bz2\n",
      "Extracting libgcc-ng-12.2.0-h65d4601_19.tar.bz2\n",
      "Extracting bzip2-1.0.8-h7f98852_4.tar.bz2\n",
      "Extracting c-ares-1.18.1-h7f98852_0.tar.bz2\n",
      "Extracting fmt-9.1.0-h924138e_0.tar.bz2\n",
      "Extracting icu-70.1-h27087fc_0.tar.bz2\n",
      "Extracting keyutils-1.6.1-h166bdaf_0.tar.bz2\n",
      "Extracting libev-4.33-h516909a_1.tar.bz2\n",
      "Extracting libffi-3.4.2-h7f98852_5.tar.bz2\n",
      "Extracting libiconv-1.17-h166bdaf_0.tar.bz2\n",
      "Extracting libnsl-2.0.0-h7f98852_0.tar.bz2\n",
      "Extracting libuuid-2.32.1-h7f98852_1000.tar.bz2\n",
      "Extracting libzlib-1.2.13-h166bdaf_4.tar.bz2\n",
      "Extracting lz4-c-1.9.3-h9c3ff4c_1.tar.bz2\n",
      "Extracting lzo-2.10-h516909a_1000.tar.bz2\n",
      "Extracting ncurses-6.3-h27087fc_1.tar.bz2\n",
      "Extracting openssl-3.0.7-h166bdaf_0.tar.bz2\n",
      "Extracting reproc-14.2.3-h7f98852_0.tar.bz2\n",
      "Extracting xz-5.2.6-h166bdaf_0.tar.bz2\n",
      "Extracting yaml-0.2.5-h7f98852_2.tar.bz2\n",
      "Extracting yaml-cpp-0.7.0-h27087fc_2.tar.bz2\n",
      "Extracting libedit-3.1.20191231-he28a2e2_2.tar.bz2\n",
      "Extracting libnghttp2-1.47.0-hff17c54_1.tar.bz2\n",
      "Extracting libsolv-0.7.22-h6239696_0.tar.bz2\n",
      "Extracting libsqlite-3.40.0-h753d276_0.tar.bz2\n",
      "Extracting libssh2-1.10.0-hf14f497_3.tar.bz2\n",
      "Extracting libxml2-2.10.3-h7463322_0.tar.bz2\n",
      "Extracting readline-8.1.2-h0f457ee_0.tar.bz2\n",
      "Extracting reproc-cpp-14.2.3-h9c3ff4c_0.tar.bz2\n",
      "Extracting tk-8.6.12-h27826a3_0.tar.bz2\n",
      "Extracting zstd-1.5.2-h6239696_4.tar.bz2\n",
      "Extracting krb5-1.19.3-h08a2579_0.tar.bz2\n",
      "Extracting libarchive-3.5.2-hada088e_3.tar.bz2\n",
      "Extracting certifi-2022.9.24-pyhd8ed1ab_0.tar.bz2\n",
      "Extracting charset-normalizer-2.1.1-pyhd8ed1ab_0.tar.bz2\n",
      "Extracting colorama-0.4.6-pyhd8ed1ab_0.tar.bz2\n",
      "Extracting idna-3.4-pyhd8ed1ab_0.tar.bz2\n",
      "Extracting libcurl-7.86.0-h2283fc2_1.tar.bz2\n",
      "Extracting pycparser-2.21-pyhd8ed1ab_0.tar.bz2\n",
      "Extracting pysocks-1.7.1-pyha2e5f31_6.tar.bz2\n",
      "Extracting python_abi-3.10-2_cp310.tar.bz2\n",
      "Extracting setuptools-65.5.1-pyhd8ed1ab_0.tar.bz2\n",
      "Extracting toolz-0.12.0-pyhd8ed1ab_0.tar.bz2\n",
      "Extracting wheel-0.38.4-pyhd8ed1ab_0.tar.bz2\n",
      "Extracting cffi-1.15.1-py310h255011f_2.tar.bz2\n",
      "Extracting libmamba-1.0.0-h9eff5f0_2.tar.bz2\n",
      "Extracting pip-22.3.1-pyhd8ed1ab_0.tar.bz2\n",
      "Extracting pycosat-0.6.4-py310h5764c6d_1.tar.bz2\n",
      "Extracting ruamel_yaml-0.15.80-py310h5764c6d_1008.tar.bz2\n",
      "Extracting tqdm-4.64.1-pyhd8ed1ab_0.tar.bz2\n",
      "Extracting brotlipy-0.7.0-py310h5764c6d_1005.tar.bz2\n",
      "Extracting conda-package-handling-1.9.0-py310h5764c6d_1.tar.bz2\n",
      "Extracting cryptography-38.0.3-py310h600f1e7_0.tar.bz2\n",
      "Extracting libmambapy-1.0.0-py310hd5a56e8_2.tar.bz2\n",
      "Extracting pyopenssl-22.1.0-pyhd8ed1ab_0.tar.bz2\n",
      "Extracting urllib3-1.26.11-pyhd8ed1ab_0.tar.bz2\n",
      "Extracting requests-2.28.1-pyhd8ed1ab_1.tar.bz2\n",
      "Extracting conda-22.9.0-py310hff52083_2.tar.bz2\n",
      "Extracting mamba-1.0.0-py310h51d5547_2.tar.bz2\n",
      "\n",
      "                                           __\n",
      "          __  ______ ___  ____ _____ ___  / /_  ____ _\n",
      "         / / / / __ `__ \\/ __ `/ __ `__ \\/ __ \\/ __ `/\n",
      "        / /_/ / / / / / / /_/ / / / / / / /_/ / /_/ /\n",
      "       / .___/_/ /_/ /_/\\__,_/_/ /_/ /_/_.___/\\__,_/\n",
      "      /_/\n",
      "\n",
      "conda-forge/linux-64                                        Using cache\n",
      "conda-forge/noarch                                          Using cache\n",
      "\n",
      "Transaction\n",
      "\n",
      "  Prefix: /home/jupyter/mambaforge\n",
      "\n",
      "  All requested packages already installed\n",
      "\n",
      "\n",
      "Transaction starting\n",
      "\u001b[?25l\u001b[2K\u001b[0G\u001b[?25hTransaction finished\n",
      "installation finished.\n",
      "\n",
      "                  __    __    __    __\n",
      "                 /  \\  /  \\  /  \\  /  \\\n",
      "                /    \\/    \\/    \\/    \\\n",
      "███████████████/  /██/  /██/  /██/  /████████████████████████\n",
      "              /  / \\   / \\   / \\   / \\  \\____\n",
      "             /  /   \\_/   \\_/   \\_/   \\    o \\__,\n",
      "            / _/                       \\_____/  `\n",
      "            |/\n",
      "        ███╗   ███╗ █████╗ ███╗   ███╗██████╗  █████╗\n",
      "        ████╗ ████║██╔══██╗████╗ ████║██╔══██╗██╔══██╗\n",
      "        ██╔████╔██║███████║██╔████╔██║██████╔╝███████║\n",
      "        ██║╚██╔╝██║██╔══██║██║╚██╔╝██║██╔══██╗██╔══██║\n",
      "        ██║ ╚═╝ ██║██║  ██║██║ ╚═╝ ██║██████╔╝██║  ██║\n",
      "        ╚═╝     ╚═╝╚═╝  ╚═╝╚═╝     ╚═╝╚═════╝ ╚═╝  ╚═╝\n",
      "\n",
      "        mamba (1.0.0) supported by @QuantStack\n",
      "\n",
      "        GitHub:  https://github.com/mamba-org/mamba\n",
      "        Twitter: https://twitter.com/QuantStack\n",
      "\n",
      "█████████████████████████████████████████████████████████████\n",
      "\n",
      "\n",
      "Looking for: ['trimmomatic', 'fastqc', 'bowtie', 'samtools', 'rsem', 'multiqc']\n",
      "\n",
      "bioconda/linux-64                                           Using cache\n",
      "bioconda/noarch                                             Using cache\n",
      "\u001b[?25l\u001b[2K\u001b[0G[+] 0.0s\n",
      "\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 0.1s\n",
      "conda-forge/linux-64 \u001b[33m━━━━━━━━━━━╸\u001b[0m\u001b[90m━━━━━━━━━━━\u001b[0m   0.0 B /  ??.?MB @  ??.?MB/s  0.1s\n",
      "conda-forge/noarch   \u001b[90m━━━━━━━━━━━━╸\u001b[0m\u001b[33m━━━━━━━━━━\u001b[0m   0.0 B /  ??.?MB @  ??.?MB/s  0.1s\n",
      "pkgs/main/linux-64   \u001b[90m━━━━━━╸\u001b[0m\u001b[33m━━━━━━━━━━━━━━━━\u001b[0m   0.0 B /  ??.?MB @  ??.?MB/s  0.1s\n",
      "pkgs/main/noarch     \u001b[90m━━━━━━━╸\u001b[0m\u001b[33m━━━━━━━━━━━━━━━\u001b[0m   0.0 B /  ??.?MB @  ??.?MB/s  0.1s\n",
      "pkgs/r/linux-64      \u001b[90m╸\u001b[0m\u001b[33m━━━━━━━━━━━━━━━╸\u001b[0m\u001b[90m━━━━━━\u001b[0m   0.0 B /  ??.?MB @  ??.?MB/s  0.1s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0Gpkgs/r/linux-64                                               No change\n",
      "pkgs/main/linux-64                                            No change\n",
      "pkgs/main/noarch                                              No change\n",
      "pkgs/r/noarch                                                 No change\n",
      "[+] 0.2s\n",
      "conda-forge/linux-64 \u001b[33m━━━━━━━━━━━━━╸\u001b[0m\u001b[90m━━━━━━━━━\u001b[0m 278.5kB /  ??.?MB @   1.5MB/s  0.2s\n",
      "conda-forge/noarch   \u001b[33m━━━━━━━━╸\u001b[0m\u001b[90m━━━━━━━━━━━━━━\u001b[0m   0.0 B /  ??.?MB @  ??.?MB/s  0.2s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 0.3s\n",
      "conda-forge/linux-64 \u001b[33m━━━━━━━━━━━━━━━╸\u001b[0m\u001b[90m━━━━━━━\u001b[0m 752.9kB /  ??.?MB @   2.7MB/s  0.3s\n",
      "conda-forge/noarch   \u001b[33m━━━━━━━━━━━╸\u001b[0m\u001b[90m━━━━━━━━━━━\u001b[0m 424.6kB /  ??.?MB @   1.7MB/s  0.3s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 0.4s\n",
      "conda-forge/linux-64 \u001b[90m━━╸\u001b[0m\u001b[33m━━━━━━━━━━━━━━━╸\u001b[0m\u001b[90m━━━━\u001b[0m   1.2MB /  ??.?MB @   3.2MB/s  0.4s\n",
      "conda-forge/noarch   \u001b[33m━━━━━━━━━━━━━╸\u001b[0m\u001b[90m━━━━━━━━━\u001b[0m 876.4kB /  ??.?MB @   2.5MB/s  0.4s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 0.5s\n",
      "conda-forge/linux-64 \u001b[90m━━━━╸\u001b[0m\u001b[33m━━━━━━━━━━━━━━━╸\u001b[0m\u001b[90m━━\u001b[0m   1.7MB /  ??.?MB @   3.6MB/s  0.5s\n",
      "conda-forge/noarch   \u001b[33m━━━━━━━━━━━━━━━╸\u001b[0m\u001b[90m━━━━━━━\u001b[0m   1.2MB /  ??.?MB @   2.7MB/s  0.5s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 0.6s\n",
      "conda-forge/linux-64 \u001b[90m━━━━━━╸\u001b[0m\u001b[33m━━━━━━━━━━━━━━━━\u001b[0m   2.3MB /  ??.?MB @   3.9MB/s  0.6s\n",
      "conda-forge/noarch   \u001b[90m━╸\u001b[0m\u001b[33m━━━━━━━━━━━━━━━╸\u001b[0m\u001b[90m━━━━━\u001b[0m   1.7MB /  ??.?MB @   3.1MB/s  0.6s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 0.7s\n",
      "conda-forge/linux-64 \u001b[90m━━━━━━━━╸\u001b[0m\u001b[33m━━━━━━━━━━━━━━\u001b[0m   2.8MB /  ??.?MB @   4.1MB/s  0.7s\n",
      "conda-forge/noarch   \u001b[90m━━━╸\u001b[0m\u001b[33m━━━━━━━━━━━━━━━╸\u001b[0m\u001b[90m━━━\u001b[0m   2.1MB /  ??.?MB @   3.3MB/s  0.7s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 0.8s\n",
      "conda-forge/linux-64 \u001b[90m━━━━━━━━━━╸\u001b[0m\u001b[33m━━━━━━━━━━━━\u001b[0m   3.3MB /  ??.?MB @   4.2MB/s  0.8s\n",
      "conda-forge/noarch   \u001b[90m━━━━━━╸\u001b[0m\u001b[33m━━━━━━━━━━━━━━━━\u001b[0m   2.5MB /  ??.?MB @   3.4MB/s  0.8s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 0.9s\n",
      "conda-forge/linux-64 \u001b[90m━━━━━━━━━━━━━╸\u001b[0m\u001b[33m━━━━━━━━━\u001b[0m   3.8MB /  ??.?MB @   4.3MB/s  0.9s\n",
      "conda-forge/noarch   \u001b[90m━━━━━━━━╸\u001b[0m\u001b[33m━━━━━━━━━━━━━━\u001b[0m   3.0MB /  ??.?MB @   3.5MB/s  0.9s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 1.0s\n",
      "conda-forge/linux-64 \u001b[33m━━━━━━━━━╸\u001b[0m\u001b[90m━━━━━━━━━━━━━\u001b[0m   4.3MB /  ??.?MB @   4.4MB/s  1.0s\n",
      "conda-forge/noarch   \u001b[90m━━━━━━━━━━╸\u001b[0m\u001b[33m━━━━━━━━━━━━\u001b[0m   3.4MB /  ??.?MB @   3.6MB/s  1.0s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 1.1s\n",
      "conda-forge/linux-64 \u001b[33m━━━━━━━━━━━╸\u001b[0m\u001b[90m━━━━━━━━━━━\u001b[0m   4.7MB /  ??.?MB @   4.3MB/s  1.1s\n",
      "conda-forge/noarch   \u001b[90m━━━━━━━━━━━━╸\u001b[0m\u001b[33m━━━━━━━━━━\u001b[0m   3.9MB /  ??.?MB @   3.7MB/s  1.1s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 1.2s\n",
      "conda-forge/linux-64 \u001b[33m━━━━━━━━━━━━━╸\u001b[0m\u001b[90m━━━━━━━━━\u001b[0m   5.2MB /  ??.?MB @   4.4MB/s  1.2s\n",
      "conda-forge/noarch   \u001b[33m━━━━━━━━╸\u001b[0m\u001b[90m━━━━━━━━━━━━━━\u001b[0m   4.4MB /  ??.?MB @   3.8MB/s  1.2s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 1.3s\n",
      "conda-forge/linux-64 \u001b[33m━━━━━━━━━━━━━━━╸\u001b[0m\u001b[90m━━━━━━━\u001b[0m   5.7MB /  ??.?MB @   4.4MB/s  1.3s\n",
      "conda-forge/noarch   \u001b[33m━━━━━━━━━━━╸\u001b[0m\u001b[90m━━━━━━━━━━━\u001b[0m   4.9MB /  ??.?MB @   3.9MB/s  1.3s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 1.4s\n",
      "conda-forge/linux-64 \u001b[90m━━╸\u001b[0m\u001b[33m━━━━━━━━━━━━━━━╸\u001b[0m\u001b[90m━━━━\u001b[0m   6.2MB /  ??.?MB @   4.5MB/s  1.4s\n",
      "conda-forge/noarch   \u001b[33m━━━━━━━━━━━━━╸\u001b[0m\u001b[90m━━━━━━━━━\u001b[0m   5.3MB /  ??.?MB @   3.9MB/s  1.4s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 1.5s\n",
      "conda-forge/linux-64 \u001b[90m━━━━╸\u001b[0m\u001b[33m━━━━━━━━━━━━━━━╸\u001b[0m\u001b[90m━━\u001b[0m   6.7MB /  ??.?MB @   4.5MB/s  1.5s\n",
      "conda-forge/noarch   \u001b[33m━━━━━━━━━━━━━━━╸\u001b[0m\u001b[90m━━━━━━━\u001b[0m   5.7MB /  ??.?MB @   3.9MB/s  1.5s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 1.6s\n",
      "conda-forge/linux-64 \u001b[90m━━━━━━╸\u001b[0m\u001b[33m━━━━━━━━━━━━━━━━\u001b[0m   7.2MB /  ??.?MB @   4.6MB/s  1.6s\n",
      "conda-forge/noarch   \u001b[90m━╸\u001b[0m\u001b[33m━━━━━━━━━━━━━━━╸\u001b[0m\u001b[90m━━━━━\u001b[0m   6.2MB /  ??.?MB @   4.0MB/s  1.6s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 1.7s\n",
      "conda-forge/linux-64 \u001b[90m━━━━━━━━╸\u001b[0m\u001b[33m━━━━━━━━━━━━━━\u001b[0m   7.7MB /  ??.?MB @   4.6MB/s  1.7s\n",
      "conda-forge/noarch   \u001b[90m━━━╸\u001b[0m\u001b[33m━━━━━━━━━━━━━━━╸\u001b[0m\u001b[90m━━━\u001b[0m   6.7MB /  ??.?MB @   4.0MB/s  1.7s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 1.8s\n",
      "conda-forge/linux-64 \u001b[90m━━━━━━━━━━╸\u001b[0m\u001b[33m━━━━━━━━━━━━\u001b[0m   8.2MB /  ??.?MB @   4.6MB/s  1.8s\n",
      "conda-forge/noarch   \u001b[90m━━━━━━╸\u001b[0m\u001b[33m━━━━━━━━━━━━━━━━\u001b[0m   7.2MB /  ??.?MB @   4.1MB/s  1.8s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 1.9s\n",
      "conda-forge/linux-64 \u001b[90m━━━━━━━━━━━━━╸\u001b[0m\u001b[33m━━━━━━━━━\u001b[0m   8.7MB /  ??.?MB @   4.6MB/s  1.9s\n",
      "conda-forge/noarch   \u001b[90m━━━━━━━━╸\u001b[0m\u001b[33m━━━━━━━━━━━━━━\u001b[0m   7.7MB /  ??.?MB @   4.1MB/s  1.9s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 2.0s\n",
      "conda-forge/linux-64 \u001b[33m━━━━━━━━━╸\u001b[0m\u001b[90m━━━━━━━━━━━━━\u001b[0m   9.2MB /  ??.?MB @   4.6MB/s  2.0s\n",
      "conda-forge/noarch   \u001b[90m━━━━━━━━━━╸\u001b[0m\u001b[33m━━━━━━━━━━━━\u001b[0m   8.1MB /  ??.?MB @   4.1MB/s  2.0s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 2.1s\n",
      "conda-forge/linux-64 \u001b[33m━━━━━━━━━━━╸\u001b[0m\u001b[90m━━━━━━━━━━━\u001b[0m   9.7MB /  ??.?MB @   4.6MB/s  2.1s\n",
      "conda-forge/noarch   \u001b[90m━━━━━━━━━━━━╸\u001b[0m\u001b[33m━━━━━━━━━━\u001b[0m   8.6MB /  ??.?MB @   4.2MB/s  2.1s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 2.2s\n",
      "conda-forge/linux-64 \u001b[33m━━━━━━━━━━━━━╸\u001b[0m\u001b[90m━━━━━━━━━\u001b[0m  10.1MB /  ??.?MB @   4.6MB/s  2.2s\n",
      "conda-forge/noarch   \u001b[33m━━━━━━━━╸\u001b[0m\u001b[90m━━━━━━━━━━━━━━\u001b[0m   9.0MB /  ??.?MB @   4.2MB/s  2.2s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 2.3s\n",
      "conda-forge/linux-64 \u001b[33m━━━━━━━━━━━━━━━╸\u001b[0m\u001b[90m━━━━━━━\u001b[0m  10.5MB /  ??.?MB @   4.6MB/s  2.3s\n",
      "conda-forge/noarch   \u001b[33m━━━━━━━━━━━╸\u001b[0m\u001b[90m━━━━━━━━━━━\u001b[0m   9.5MB /  ??.?MB @   4.2MB/s  2.3s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 2.4s\n",
      "conda-forge/linux-64 \u001b[90m━━╸\u001b[0m\u001b[33m━━━━━━━━━━━━━━━╸\u001b[0m\u001b[90m━━━━\u001b[0m  11.1MB /  ??.?MB @   4.6MB/s  2.4s\n",
      "conda-forge/noarch   \u001b[33m━━━━━━━━━━━━━╸\u001b[0m\u001b[90m━━━━━━━━━\u001b[0m  10.0MB /  ??.?MB @   4.2MB/s  2.4s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 2.5s\n",
      "conda-forge/linux-64 \u001b[90m━━━━╸\u001b[0m\u001b[33m━━━━━━━━━━━━━━━╸\u001b[0m\u001b[90m━━\u001b[0m  11.5MB /  ??.?MB @   4.6MB/s  2.5s\n",
      "conda-forge/noarch   \u001b[33m━━━━━━━━━━━━━━━╸\u001b[0m\u001b[90m━━━━━━━\u001b[0m  10.4MB /  ??.?MB @   4.2MB/s  2.5s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 2.6s\n",
      "conda-forge/linux-64 \u001b[90m━━━━━╸\u001b[0m\u001b[33m━━━━━━━━━━━━━━━╸\u001b[0m\u001b[90m━\u001b[0m  11.8MB /  ??.?MB @   4.6MB/s  2.6s\n",
      "conda-forge/noarch   \u001b[90m╸\u001b[0m\u001b[33m━━━━━━━━━━━━━━━╸\u001b[0m\u001b[90m━━━━━━\u001b[0m  10.6MB /  ??.?MB @   4.2MB/s  2.6s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 2.7s\n",
      "conda-forge/linux-64 \u001b[90m━━━━━╸\u001b[0m\u001b[33m━━━━━━━━━━━━━━━╸\u001b[0m\u001b[90m━\u001b[0m  11.8MB /  ??.?MB @   4.6MB/s  2.7s\n",
      "conda-forge/noarch   \u001b[90m╸\u001b[0m\u001b[33m━━━━━━━━━━━━━━━╸\u001b[0m\u001b[90m━━━━━━\u001b[0m  10.6MB /  ??.?MB @   4.2MB/s  2.7s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 2.8s\n",
      "conda-forge/linux-64 \u001b[90m━━━━━╸\u001b[0m\u001b[33m━━━━━━━━━━━━━━━╸\u001b[0m\u001b[90m━\u001b[0m  11.8MB /  ??.?MB @   4.6MB/s  2.8s\n",
      "conda-forge/noarch   \u001b[90m╸\u001b[0m\u001b[33m━━━━━━━━━━━━━━━╸\u001b[0m\u001b[90m━━━━━━\u001b[0m  10.6MB /  ??.?MB @   4.2MB/s  2.8s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 2.9s\n",
      "conda-forge/linux-64 \u001b[90m━━━━━╸\u001b[0m\u001b[33m━━━━━━━━━━━━━━━╸\u001b[0m\u001b[90m━\u001b[0m  11.8MB /  ??.?MB @   4.6MB/s  2.9s\n",
      "conda-forge/noarch   \u001b[90m╸\u001b[0m\u001b[33m━━━━━━━━━━━━━━━╸\u001b[0m\u001b[90m━━━━━━\u001b[0m  10.6MB /  ??.?MB @   4.2MB/s  2.9s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 3.0s\n",
      "conda-forge/linux-64 \u001b[90m━━━━╸\u001b[0m\u001b[33m━━━━━━━━━━━━━━━╸\u001b[0m\u001b[90m━\u001b[0m  11.8MB @   4.6MB/s             3.0s\n",
      "conda-forge/noarch   ━━━━━━━━━━━━━━━━━━━━━━  10.7MB @   4.2MB/s Finalizing  3.0s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 3.1s\n",
      "conda-forge/linux-64 \u001b[90m━━━━━╸\u001b[0m\u001b[33m━━━━━━━━━━━━━━━╸\u001b[0m\u001b[90m━\u001b[0m  11.8MB /  ??.?MB @   4.6MB/s  3.1s\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 3.2s\n",
      "conda-forge/linux-64 \u001b[90m━━━━━╸\u001b[0m\u001b[33m━━━━━━━━━━━━━━━╸\u001b[0m\u001b[90m━\u001b[0m  11.8MB /  ??.?MB @   4.6MB/s  3.2s\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 3.3s\n",
      "conda-forge/linux-64 \u001b[90m━━━━━╸\u001b[0m\u001b[33m━━━━━━━━━━━━━━━╸\u001b[0m\u001b[90m━\u001b[0m  11.8MB /  ??.?MB @   4.6MB/s  3.3s\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 3.4s\n",
      "conda-forge/linux-64 \u001b[90m━━━━━╸\u001b[0m\u001b[33m━━━━━━━━━━━━━━━╸\u001b[0m\u001b[90m━\u001b[0m  11.8MB /  ??.?MB @   4.6MB/s  3.4s\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 3.5s\n",
      "conda-forge/linux-64 \u001b[90m━━━━━╸\u001b[0m\u001b[33m━━━━━━━━━━━━━━━╸\u001b[0m\u001b[90m━\u001b[0m  11.8MB /  ??.?MB @   4.6MB/s  3.5s\u001b[2K\u001b[1A\u001b[2K\u001b[0Gconda-forge/noarch                                 @   4.2MB/s  3.0s\n",
      "[+] 3.6s\n",
      "conda-forge/linux-64 \u001b[90m━━━━━━╸\u001b[0m\u001b[33m━━━━━━━━━━━━━━━━\u001b[0m  11.8MB /  ??.?MB @   3.3MB/s  3.6s\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 3.7s\n",
      "conda-forge/linux-64 \u001b[90m━━━━━━━━╸\u001b[0m\u001b[33m━━━━━━━━━━━━━━\u001b[0m  13.2MB /  ??.?MB @   3.6MB/s  3.7s\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 3.8s\n",
      "conda-forge/linux-64 \u001b[90m━━━━━━━━━━╸\u001b[0m\u001b[33m━━━━━━━━━━━━\u001b[0m  13.6MB /  ??.?MB @   3.6MB/s  3.8s\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 3.9s\n",
      "conda-forge/linux-64 \u001b[90m━━━━━━━━━━━━━╸\u001b[0m\u001b[33m━━━━━━━━━\u001b[0m  14.1MB /  ??.?MB @   3.6MB/s  3.9s\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 4.0s\n",
      "conda-forge/linux-64 \u001b[33m━━━━━━━━━╸\u001b[0m\u001b[90m━━━━━━━━━━━━━\u001b[0m  14.6MB /  ??.?MB @   3.7MB/s  4.0s\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 4.1s\n",
      "conda-forge/linux-64 \u001b[33m━━━━━━━━━━━╸\u001b[0m\u001b[90m━━━━━━━━━━━\u001b[0m  15.1MB /  ??.?MB @   3.7MB/s  4.1s\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 4.2s\n",
      "conda-forge/linux-64 \u001b[33m━━━━━━━━━━━━━╸\u001b[0m\u001b[90m━━━━━━━━━\u001b[0m  15.6MB /  ??.?MB @   3.7MB/s  4.2s\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 4.3s\n",
      "conda-forge/linux-64 \u001b[33m━━━━━━━━━━━━━━━╸\u001b[0m\u001b[90m━━━━━━━\u001b[0m  16.1MB /  ??.?MB @   3.8MB/s  4.3s\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 4.4s\n",
      "conda-forge/linux-64 \u001b[90m━━╸\u001b[0m\u001b[33m━━━━━━━━━━━━━━━╸\u001b[0m\u001b[90m━━━━\u001b[0m  16.7MB /  ??.?MB @   3.8MB/s  4.4s\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 4.5s\n",
      "conda-forge/linux-64 \u001b[90m━━━━╸\u001b[0m\u001b[33m━━━━━━━━━━━━━━━╸\u001b[0m\u001b[90m━━\u001b[0m  17.2MB /  ??.?MB @   3.8MB/s  4.5s\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 4.6s\n",
      "conda-forge/linux-64 \u001b[90m━━━━━━╸\u001b[0m\u001b[33m━━━━━━━━━━━━━━━━\u001b[0m  17.7MB /  ??.?MB @   3.8MB/s  4.6s\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 4.7s\n",
      "conda-forge/linux-64 \u001b[90m━━━━━━━━╸\u001b[0m\u001b[33m━━━━━━━━━━━━━━\u001b[0m  18.1MB /  ??.?MB @   3.9MB/s  4.7s\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 4.8s\n",
      "conda-forge/linux-64 \u001b[90m━━━━━━━━━━╸\u001b[0m\u001b[33m━━━━━━━━━━━━\u001b[0m  18.6MB /  ??.?MB @   3.9MB/s  4.8s\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 4.9s\n",
      "conda-forge/linux-64 \u001b[90m━━━━━━━━━━━━━╸\u001b[0m\u001b[33m━━━━━━━━━\u001b[0m  19.1MB /  ??.?MB @   3.9MB/s  4.9s\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 5.0s\n",
      "conda-forge/linux-64 \u001b[33m━━━━━━━━━╸\u001b[0m\u001b[90m━━━━━━━━━━━━━\u001b[0m  19.6MB /  ??.?MB @   3.9MB/s  5.0s\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 5.1s\n",
      "conda-forge/linux-64 \u001b[33m━━━━━━━━━━━╸\u001b[0m\u001b[90m━━━━━━━━━━━\u001b[0m  20.1MB /  ??.?MB @   3.9MB/s  5.1s\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 5.2s\n",
      "conda-forge/linux-64 \u001b[33m━━━━━━━━━━━━━╸\u001b[0m\u001b[90m━━━━━━━━━\u001b[0m  20.5MB /  ??.?MB @   4.0MB/s  5.2s\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 5.3s\n",
      "conda-forge/linux-64 \u001b[33m━━━━━━━━━━━━━━━╸\u001b[0m\u001b[90m━━━━━━━\u001b[0m  21.0MB /  ??.?MB @   4.0MB/s  5.3s\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 5.4s\n",
      "conda-forge/linux-64 \u001b[90m━━╸\u001b[0m\u001b[33m━━━━━━━━━━━━━━━╸\u001b[0m\u001b[90m━━━━\u001b[0m  21.4MB /  ??.?MB @   4.0MB/s  5.4s\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 5.5s\n",
      "conda-forge/linux-64 \u001b[90m━━━━╸\u001b[0m\u001b[33m━━━━━━━━━━━━━━━╸\u001b[0m\u001b[90m━━\u001b[0m  21.9MB /  ??.?MB @   4.0MB/s  5.5s\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 5.6s\n",
      "conda-forge/linux-64 \u001b[90m━━━━━━╸\u001b[0m\u001b[33m━━━━━━━━━━━━━━━━\u001b[0m  22.3MB /  ??.?MB @   4.0MB/s  5.6s\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 5.7s\n",
      "conda-forge/linux-64 \u001b[90m━━━━━━━━╸\u001b[0m\u001b[33m━━━━━━━━━━━━━━\u001b[0m  22.8MB /  ??.?MB @   4.0MB/s  5.7s\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 5.8s\n",
      "conda-forge/linux-64 \u001b[90m━━━━━━━━━━╸\u001b[0m\u001b[33m━━━━━━━━━━━━\u001b[0m  23.3MB /  ??.?MB @   4.0MB/s  5.8s\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 5.9s\n",
      "conda-forge/linux-64 \u001b[90m━━━━━━━━━━━━━╸\u001b[0m\u001b[33m━━━━━━━━━\u001b[0m  23.8MB /  ??.?MB @   4.0MB/s  5.9s\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 6.0s\n",
      "conda-forge/linux-64 \u001b[33m━━━━━━━━━╸\u001b[0m\u001b[90m━━━━━━━━━━━━━\u001b[0m  24.3MB /  ??.?MB @   4.0MB/s  6.0s\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 6.1s\n",
      "conda-forge/linux-64 \u001b[33m━━━━━━━━━━━╸\u001b[0m\u001b[90m━━━━━━━━━━━\u001b[0m  24.7MB /  ??.?MB @   4.1MB/s  6.1s\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 6.2s\n",
      "conda-forge/linux-64 \u001b[33m━━━━━━━━━━━━╸\u001b[0m\u001b[90m━━━━━━━━━━\u001b[0m  25.0MB /  ??.?MB @   4.1MB/s  6.2s\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 6.3s\n",
      "conda-forge/linux-64 \u001b[33m━━━━━━━━━━━━━━╸\u001b[0m\u001b[90m━━━━━━━━\u001b[0m  25.4MB /  ??.?MB @   4.1MB/s  6.3s\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 6.4s\n",
      "conda-forge/linux-64 \u001b[90m╸\u001b[0m\u001b[33m━━━━━━━━━━━━━━━╸\u001b[0m\u001b[90m━━━━━━\u001b[0m  25.8MB /  ??.?MB @   4.1MB/s  6.4s\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 6.5s\n",
      "conda-forge/linux-64 \u001b[90m━━━╸\u001b[0m\u001b[33m━━━━━━━━━━━━━━━╸\u001b[0m\u001b[90m━━━\u001b[0m  26.3MB /  ??.?MB @   4.1MB/s  6.5s\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 6.6s\n",
      "conda-forge/linux-64 \u001b[90m━━━━━╸\u001b[0m\u001b[33m━━━━━━━━━━━━━━━╸\u001b[0m\u001b[90m━\u001b[0m  26.8MB /  ??.?MB @   4.1MB/s  6.6s\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 6.7s\n",
      "conda-forge/linux-64 \u001b[90m━━━━━━━╸\u001b[0m\u001b[33m━━━━━━━━━━━━━━━\u001b[0m  27.2MB /  ??.?MB @   4.1MB/s  6.7s\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 6.8s\n",
      "conda-forge/linux-64 \u001b[90m━━━━━━━━━╸\u001b[0m\u001b[33m━━━━━━━━━━━━━\u001b[0m  27.7MB /  ??.?MB @   4.1MB/s  6.8s\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 6.9s\n",
      "conda-forge/linux-64 \u001b[90m━━━━━━━━━━━╸\u001b[0m\u001b[33m━━━━━━━━━━━\u001b[0m  28.1MB /  ??.?MB @   4.1MB/s  6.9s\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 7.0s\n",
      "conda-forge/linux-64 \u001b[33m━━━━━━━━╸\u001b[0m\u001b[90m━━━━━━━━━━━━━━\u001b[0m  28.5MB /  ??.?MB @   4.1MB/s  7.0s\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 7.1s\n",
      "conda-forge/linux-64 \u001b[33m━━━━━━━━━╸\u001b[0m\u001b[90m━━━━━━━━━━━━━\u001b[0m  28.7MB /  ??.?MB @   4.1MB/s  7.1s\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 7.2s\n",
      "conda-forge/linux-64 \u001b[33m━━━━━━━━━╸\u001b[0m\u001b[90m━━━━━━━━━━━━━\u001b[0m  28.7MB /  ??.?MB @   4.1MB/s  7.2s\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 7.3s\n",
      "conda-forge/linux-64 \u001b[33m━━━━━━━━━╸\u001b[0m\u001b[90m━━━━━━━━━━━━━\u001b[0m  28.7MB /  ??.?MB @   4.1MB/s  7.3s\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 7.4s\n",
      "conda-forge/linux-64 \u001b[33m━━━━━━━━━╸\u001b[0m\u001b[90m━━━━━━━━━━━━━\u001b[0m  28.7MB /  ??.?MB @   4.1MB/s  7.4s\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 7.5s\n",
      "conda-forge/linux-64 \u001b[33m━━━━━━━━━╸\u001b[0m\u001b[90m━━━━━━━━━━━━━\u001b[0m  28.7MB /  ??.?MB @   4.1MB/s  7.5s\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 7.6s\n",
      "conda-forge/linux-64 \u001b[33m━━━━━━━━━╸\u001b[0m\u001b[90m━━━━━━━━━━━━━\u001b[0m  28.7MB /  ??.?MB @   4.1MB/s  7.6s\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 7.7s\n",
      "conda-forge/linux-64 \u001b[33m━━━━━━━━━╸\u001b[0m\u001b[90m━━━━━━━━━━━━━\u001b[0m  28.7MB /  ??.?MB @   4.1MB/s  7.7s\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 7.8s\n",
      "conda-forge/linux-64 \u001b[33m━━━━━━━━━╸\u001b[0m\u001b[90m━━━━━━━━━━━━━\u001b[0m  28.7MB /  ??.?MB @   4.1MB/s  7.8s\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 7.9s\n",
      "conda-forge/linux-64 \u001b[33m━━━━━━━━━╸\u001b[0m\u001b[90m━━━━━━━━━━━━━\u001b[0m  28.7MB /  ??.?MB @   4.1MB/s  7.9s\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 8.0s\n",
      "conda-forge/linux-64 \u001b[33m━━━━━━━━━╸\u001b[0m\u001b[90m━━━━━━━━━━━━━\u001b[0m  28.7MB /  ??.?MB @   4.1MB/s  8.0s\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 8.1s\n",
      "conda-forge/linux-64 \u001b[33m━━━━━━━━━╸\u001b[0m\u001b[90m━━━━━━━━━━━━━\u001b[0m  28.7MB /  ??.?MB @   4.1MB/s  8.1s\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 8.2s\n",
      "conda-forge/linux-64 \u001b[33m━━━━━━━━━╸\u001b[0m\u001b[90m━━━━━━━━━━━━━\u001b[0m  28.7MB /  ??.?MB @   4.1MB/s  8.2s\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 8.3s\n",
      "conda-forge/linux-64 \u001b[33m━━━━━━━━━╸\u001b[0m\u001b[90m━━━━━━━━━━━━━\u001b[0m  28.7MB /  ??.?MB @   4.1MB/s  8.3s\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 8.4s\n",
      "conda-forge/linux-64 \u001b[33m━━━━━━━━━╸\u001b[0m\u001b[90m━━━━━━━━━━━━━\u001b[0m  28.7MB /  ??.?MB @   4.1MB/s  8.4s\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 8.5s\n",
      "conda-forge/linux-64 \u001b[33m━━━━━━━━━╸\u001b[0m\u001b[90m━━━━━━━━━━━━━\u001b[0m  28.7MB /  ??.?MB @   4.1MB/s  8.5s\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 8.6s\n",
      "conda-forge/linux-64 \u001b[33m━━━━━━━━━╸\u001b[0m\u001b[90m━━━━━━━━━━━━━\u001b[0m  28.7MB /  ??.?MB @   4.1MB/s  8.6s\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 8.7s\n",
      "conda-forge/linux-64 \u001b[33m━━━━━━━━━╸\u001b[0m\u001b[90m━━━━━━━━━━━━━\u001b[0m  28.7MB /  ??.?MB @   4.1MB/s  8.7s\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 8.8s\n",
      "conda-forge/linux-64 \u001b[33m━━━━━━━━━╸\u001b[0m\u001b[90m━━━━━━━━━━━━━\u001b[0m  28.7MB /  ??.?MB @   4.1MB/s  8.8s\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 8.9s\n",
      "conda-forge/linux-64 ━━━━━━━━━━━━━━━━━━━━━━  28.7MB @   4.1MB/s Finalizing  8.9s\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 9.0s\n",
      "\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 9.1s\n",
      "\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 9.2s\n",
      "\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 9.3s\n",
      "\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 9.4s\n",
      "\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 9.5s\n",
      "\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 9.6s\n",
      "\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 9.7s\n",
      "\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 9.8s\n",
      "\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 9.9s\n",
      "\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 10.0s\n",
      "\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 10.1s\n",
      "\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 10.2s\n",
      "\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 10.3s\n",
      "\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 10.4s\n",
      "\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 10.5s\n",
      "\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 10.6s\n",
      "\u001b[2K\u001b[1A\u001b[2K\u001b[0Gconda-forge/linux-64                               @   4.1MB/s  9.0s\n",
      "\u001b[?25h\n",
      "Pinned packages:\n",
      "  - python 3.7.*\n",
      "\n",
      "\n",
      "Transaction\n",
      "\n",
      "  Prefix: /opt/conda\n",
      "\n",
      "  Updating specs:\n",
      "\n",
      "   - trimmomatic\n",
      "   - fastqc\n",
      "   - bowtie\n",
      "   - samtools\n",
      "   - rsem\n",
      "   - multiqc\n",
      "   - ca-certificates\n",
      "   - certifi\n",
      "   - openssl\n",
      "\n",
      "\n",
      "  Package              Version  Build             Channel                   Size\n",
      "──────────────────────────────────────────────────────────────────────────────────\n",
      "  Install:\n",
      "──────────────────────────────────────────────────────────────────────────────────\n",
      "\n",
      "\u001b[32m  + bottleneck       \u001b[00m    1.3.5  py37hda87dfa_0    conda-forge/linux-64     129kB\n",
      "\u001b[32m  + brotli           \u001b[00m    1.0.9  h166bdaf_8        conda-forge/linux-64      19kB\n",
      "\u001b[32m  + brotli-bin       \u001b[00m    1.0.9  h166bdaf_8        conda-forge/linux-64      20kB\n",
      "\u001b[32m  + click            \u001b[00m    8.1.3  py37h89c1867_0    conda-forge/linux-64     148kB\n",
      "\u001b[32m  + coloredlogs      \u001b[00m   15.0.1  pyhd8ed1ab_3      conda-forge/noarch        41kB\n",
      "\u001b[32m  + colormath        \u001b[00m    3.0.0  py_2              conda-forge/noarch        35kB\n",
      "\u001b[32m  + commonmark       \u001b[00m    0.9.1  py_0              conda-forge/noarch        47kB\n",
      "\u001b[32m  + cycler           \u001b[00m   0.11.0  pyhd8ed1ab_0      conda-forge/noarch        10kB\n",
      "\u001b[32m  + dataclasses      \u001b[00m      0.8  pyhc8e2a94_3      conda-forge/noarch        10kB\n",
      "\u001b[32m  + fonttools        \u001b[00m   4.38.0  py37h540881e_0    conda-forge/linux-64       2MB\n",
      "\u001b[32m  + future           \u001b[00m   0.18.2  py37h89c1867_5    conda-forge/linux-64     730kB\n",
      "\u001b[32m  + humanfriendly    \u001b[00m     10.0  py37h89c1867_2    conda-forge/linux-64     121kB\n",
      "\u001b[32m  + kiwisolver       \u001b[00m    1.4.4  py37h7cecad7_0    conda-forge/linux-64      75kB\n",
      "\u001b[32m  + libbrotlicommon  \u001b[00m    1.0.9  h166bdaf_8        conda-forge/linux-64      67kB\n",
      "\u001b[32m  + libbrotlidec     \u001b[00m    1.0.9  h166bdaf_8        conda-forge/linux-64      34kB\n",
      "\u001b[32m  + libbrotlienc     \u001b[00m    1.0.9  h166bdaf_8        conda-forge/linux-64     295kB\n",
      "\u001b[32m  + lzstring         \u001b[00m    1.0.4  py_1001           conda-forge/noarch         6kB\n",
      "\u001b[32m  + markdown         \u001b[00m    3.4.1  pyhd8ed1ab_0      conda-forge/noarch        66kB\n",
      "\u001b[32m  + matplotlib-base  \u001b[00m    3.5.3  py37hf395dca_2    conda-forge/linux-64       8MB\n",
      "\u001b[32m  + multiqc          \u001b[00m     1.13  pyhdfd78af_0      bioconda/noarch          973kB\n",
      "\u001b[32m  + munkres          \u001b[00m    1.1.4  pyh9f0ad1d_0      conda-forge/noarch        12kB\n",
      "\u001b[32m  + networkx         \u001b[00m      2.7  pyhd8ed1ab_0      conda-forge/noarch         2MB\n",
      "\u001b[32m  + nomkl            \u001b[00m      1.0  h5ca1d4c_0        conda-forge/noarch         4kB\n",
      "\u001b[32m  + numexpr          \u001b[00m    2.8.3  py37h85a3170_100  conda-forge/linux-64     137kB\n",
      "\u001b[32m  + numpy            \u001b[00m   1.21.6  py37h976b520_0    conda-forge/linux-64       6MB\n",
      "\u001b[32m  + openjpeg         \u001b[00m    2.5.0  h7d73246_1        conda-forge/linux-64     546kB\n",
      "\u001b[32m  + pandas           \u001b[00m    1.3.5  py37h8c16a72_0    pkgs/main/linux-64        10MB\n",
      "\u001b[32m  + pillow           \u001b[00m    9.2.0  py37h850a105_2    conda-forge/linux-64      47MB\n",
      "\u001b[32m  + pyparsing        \u001b[00m    3.0.9  pyhd8ed1ab_0      conda-forge/noarch        81kB\n",
      "\u001b[32m  + pytz             \u001b[00m   2022.7  pyhd8ed1ab_0      conda-forge/noarch       186kB\n",
      "\u001b[32m  + pyyaml           \u001b[00m      6.0  py37h540881e_4    conda-forge/linux-64     182kB\n",
      "\u001b[32m  + rich             \u001b[00m   13.0.0  pyhd8ed1ab_0      conda-forge/noarch       182kB\n",
      "\u001b[32m  + rich-click       \u001b[00m    1.6.0  pyhd8ed1ab_0      conda-forge/noarch        23kB\n",
      "\u001b[32m  + scipy            \u001b[00m    1.7.3  py37hf2a6cf1_0    conda-forge/linux-64      23MB\n",
      "\u001b[32m  + simplejson       \u001b[00m   3.17.6  py37h540881e_1    conda-forge/linux-64     106kB\n",
      "\u001b[32m  + spectra          \u001b[00m   0.0.11  py_1              conda-forge/noarch        19kB\n",
      "\u001b[32m  + typing-extensions\u001b[00m    4.4.0  hd8ed1ab_0        conda-forge/noarch         9kB\n",
      "\u001b[32m  + unicodedata2     \u001b[00m   14.0.0  py37h540881e_1    conda-forge/linux-64     508kB\n",
      "\n",
      "  Summary:\n",
      "\n",
      "  Install: 38 packages\n",
      "\n",
      "  Total download: 102MB\n",
      "\n",
      "──────────────────────────────────────────────────────────────────────────────────\n",
      "\n",
      "\u001b[?25l\u001b[2K\u001b[0G[+] 0.0s\n",
      "Downloading      \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m   0.0 B                            0.0s\n",
      "Extracting       \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m       0                            0.0s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 0.1s\n",
      "Downloading  (5) \u001b[33m╸\u001b[0m\u001b[90m━━━━━━━━━━━━━━━━━━━━━━\u001b[0m   0.0 B click                      0.0s\n",
      "Extracting       \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m       0                            0.0s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0Gclick                                              148.0kB @   1.2MB/s  0.1s\n",
      "simplejson                                         105.8kB @ 822.0kB/s  0.1s\n",
      "pyyaml                                             182.0kB @   1.4MB/s  0.1s\n",
      "future                                             729.9kB @   5.1MB/s  0.1s\n",
      "markdown                                            66.2kB @ 407.4kB/s  0.0s\n",
      "typing-extensions                                    8.6kB @  43.9kB/s  0.1s\n",
      "[+] 0.2s\n",
      "Downloading  (5) ╸\u001b[33m━━━━━━━━━━━━━━━━━╸\u001b[0m\u001b[90m━━━━\u001b[0m   8.4MB lzstring                   0.1s\n",
      "Extracting   (5) \u001b[33m━━━━━━━╸\u001b[0m\u001b[90m━━━━━━━━━━━━━━━\u001b[0m       1 future                     0.0s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0Gpytz                                               186.3kB @ 933.6kB/s  0.0s\n",
      "numpy                                                6.4MB @  29.3MB/s  0.2s\n",
      "lzstring                                             6.2kB @  27.5kB/s  0.0s\n",
      "rich-click                                          22.6kB @  97.5kB/s  0.0s\n",
      "numexpr                                            137.0kB @ 538.2kB/s  0.0s\n",
      "networkx                                             1.6MB @   6.0MB/s  0.0s\n",
      "unicodedata2                                       508.2kB @   1.8MB/s  0.0s\n",
      "multiqc                                            972.8kB @   3.3MB/s  0.1s\n",
      "scipy                                               22.8MB @  77.5MB/s  0.2s\n",
      "[+] 0.3s\n",
      "Downloading  (5) ━━━━━━━━╸\u001b[33m━━━━━━━━━━━━━━\u001b[0m  42.6MB brotli                     0.2s\n",
      "Extracting  (13) ╸\u001b[33m━━━━━━━━╸\u001b[0m\u001b[90m━━━━━━━━━━━━━\u001b[0m       3 future                     0.1s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0Glibbrotlienc                                       295.2kB @ 988.2kB/s  0.0s\n",
      "brotli                                              18.9kB @  60.2kB/s  0.0s\n",
      "commonmark                                          47.4kB @ 139.7kB/s  0.0s\n",
      "kiwisolver                                          74.5kB @ 217.8kB/s  0.0s\n",
      "pyparsing                                           81.3kB @ 234.0kB/s  0.1s\n",
      "bottleneck                                         129.2kB @ 352.0kB/s  0.0s\n",
      "cycler                                              10.3kB @  27.8kB/s  0.0s\n",
      "humanfriendly                                      121.3kB @ 322.4kB/s  0.0s\n",
      "pandas                                               9.7MB @  24.8MB/s  0.1s\n",
      "coloredlogs                                         40.6kB @ 103.3kB/s  0.0s\n",
      "[+] 0.4s\n",
      "Downloading  (5) ━━━━━━━━━━━╸\u001b[33m━━━━━━━━━━━\u001b[0m  54.5MB brotli-bin                 0.3s\n",
      "Extracting  (21) ━╸\u001b[33m━━━━━━━━━━━━━╸\u001b[0m\u001b[90m━━━━━━━\u001b[0m       4 bottleneck                 0.2s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0Gnomkl                                                3.8kB @   9.1kB/s  0.0s\n",
      "libbrotlicommon                                     67.2kB @ 157.7kB/s  0.1s\n",
      "brotli-bin                                          20.1kB @  43.3kB/s  0.1s\n",
      "libbrotlidec                                        34.2kB @  73.2kB/s  0.0s\n",
      "colormath                                           35.3kB @  74.7kB/s  0.1s\n",
      "dataclasses                                          9.9kB @  20.0kB/s  0.0s\n",
      "[+] 0.5s\n",
      "Downloading  (5) ━━━━━━━━━━━━━╸\u001b[33m━━━━━━━━━\u001b[0m  62.6MB fonttools                  0.4s\n",
      "Extracting  (25) ━━━╸\u001b[33m━━━━━━━━━━━━━━━╸\u001b[0m\u001b[90m━━━\u001b[0m       7 bottleneck                 0.3s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0Gspectra                                             19.4kB @  38.9kB/s  0.0s\n",
      "fonttools                                            2.0MB @   4.0MB/s  0.0s\n",
      "rich                                               182.4kB @ 353.8kB/s  0.0s\n",
      "openjpeg                                           546.2kB @   1.0MB/s  0.0s\n",
      "munkres                                             12.5kB @  23.1kB/s  0.0s\n",
      "matplotlib-base                                      7.7MB @  14.1MB/s  0.2s\n",
      "[+] 0.6s\n",
      "Downloading  (1) ━━━━━━━━━━━━━━━━╸\u001b[33m━━━━━━\u001b[0m  78.4MB pillow                     0.5s\n",
      "Extracting  (30) ━━━╸\u001b[33m━━━━━━━━━━━━━━━━━━━\u001b[0m       7 bottleneck                 0.4s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 0.7s\n",
      "Downloading  (1) ━━━━━━━━━━━━━━━━━━╸\u001b[33m━━━━\u001b[0m  87.5MB pillow                     0.6s\n",
      "Extracting  (30) ━━━╸\u001b[33m━━━━━━━━━━━━━━━━━━━\u001b[0m       7 bottleneck                 0.5s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 0.8s\n",
      "Downloading  (1) ━━━━━━━━━━━━━━━━━━━━╸\u001b[33m━━\u001b[0m  95.4MB pillow                     0.7s\n",
      "Extracting  (30) ━━━╸\u001b[33m━━━━━━━━━━━━━━━━━━━\u001b[0m       7 brotli                     0.6s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0Gpillow                                              47.2MB @  54.9MB/s  0.7s\n",
      "[+] 0.9s\n",
      "Downloading      ━━━━━━━━━━━━━━━━━━━━━━━ 102.3MB                            0.8s\n",
      "Extracting  (31) ━━━╸\u001b[33m━━━━━━━━━━━━━━━━━━━\u001b[0m       7 brotli                     0.7s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 1.0s\n",
      "Downloading      ━━━━━━━━━━━━━━━━━━━━━━━ 102.3MB                            0.8s\n",
      "Extracting  (31) ━━━╸\u001b[33m━━━━━━━━━━━━━━━━━━━\u001b[0m       7 brotli                     0.8s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 1.1s\n",
      "Downloading      ━━━━━━━━━━━━━━━━━━━━━━━ 102.3MB                            0.8s\n",
      "Extracting  (31) ━━━╸\u001b[33m━━━━━━━━━━━━━━━━━━━\u001b[0m       7 brotli                     0.9s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 1.2s\n",
      "Downloading      ━━━━━━━━━━━━━━━━━━━━━━━ 102.3MB                            0.8s\n",
      "Extracting  (27) ━━━━━╸\u001b[33m━━━━━━━━━━━━━━━━━\u001b[0m      11 brotli-bin                 1.0s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 1.3s\n",
      "Downloading      ━━━━━━━━━━━━━━━━━━━━━━━ 102.3MB                            0.8s\n",
      "Extracting  (27) ━━━━━╸\u001b[33m━━━━━━━━━━━━━━━━━\u001b[0m      11 brotli-bin                 1.1s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 1.4s\n",
      "Downloading      ━━━━━━━━━━━━━━━━━━━━━━━ 102.3MB                            0.8s\n",
      "Extracting  (26) ━━━━━━╸\u001b[33m━━━━━━━━━━━━━━━━\u001b[0m      12 brotli-bin                 1.2s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 1.5s\n",
      "Downloading      ━━━━━━━━━━━━━━━━━━━━━━━ 102.3MB                            0.8s\n",
      "Extracting  (25) ━━━━━━╸\u001b[33m━━━━━━━━━━━━━━━━\u001b[0m      13 brotli-bin                 1.3s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 1.6s\n",
      "Downloading      ━━━━━━━━━━━━━━━━━━━━━━━ 102.3MB                            0.8s\n",
      "Extracting  (23) ━━━━━━━━╸\u001b[33m━━━━━━━━━━━━━━\u001b[0m      15 coloredlogs                1.4s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 1.7s\n",
      "Downloading      ━━━━━━━━━━━━━━━━━━━━━━━ 102.3MB                            0.8s\n",
      "Extracting  (23) ━━━━━━━━╸\u001b[33m━━━━━━━━━━━━━━\u001b[0m      15 coloredlogs                1.5s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 1.8s\n",
      "Downloading      ━━━━━━━━━━━━━━━━━━━━━━━ 102.3MB                            0.8s\n",
      "Extracting  (23) ━━━━━━━━╸\u001b[33m━━━━━━━━━━━━━━\u001b[0m      15 coloredlogs                1.6s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 1.9s\n",
      "Downloading      ━━━━━━━━━━━━━━━━━━━━━━━ 102.3MB                            0.8s\n",
      "Extracting  (23) ━━━━━━━━╸\u001b[33m━━━━━━━━━━━━━━\u001b[0m      15 coloredlogs                1.7s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 2.0s\n",
      "Downloading      ━━━━━━━━━━━━━━━━━━━━━━━ 102.3MB                            0.8s\n",
      "Extracting  (23) ━━━━━━━━╸\u001b[33m━━━━━━━━━━━━━━\u001b[0m      15 colormath                  1.8s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 2.1s\n",
      "Downloading      ━━━━━━━━━━━━━━━━━━━━━━━ 102.3MB                            0.8s\n",
      "Extracting  (23) ━━━━━━━━╸\u001b[33m━━━━━━━━━━━━━━\u001b[0m      15 colormath                  1.9s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 2.2s\n",
      "Downloading      ━━━━━━━━━━━━━━━━━━━━━━━ 102.3MB                            0.8s\n",
      "Extracting  (23) ━━━━━━━━╸\u001b[33m━━━━━━━━━━━━━━\u001b[0m      15 colormath                  2.0s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 2.3s\n",
      "Downloading      ━━━━━━━━━━━━━━━━━━━━━━━ 102.3MB                            0.8s\n",
      "Extracting  (23) ━━━━━━━━╸\u001b[33m━━━━━━━━━━━━━━\u001b[0m      15 colormath                  2.1s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 2.4s\n",
      "Downloading      ━━━━━━━━━━━━━━━━━━━━━━━ 102.3MB                            0.8s\n",
      "Extracting  (23) ━━━━━━━━╸\u001b[33m━━━━━━━━━━━━━━\u001b[0m      15 commonmark                 2.2s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 2.5s\n",
      "Downloading      ━━━━━━━━━━━━━━━━━━━━━━━ 102.3MB                            0.8s\n",
      "Extracting  (23) ━━━━━━━━╸\u001b[33m━━━━━━━━━━━━━━\u001b[0m      15 commonmark                 2.3s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 2.6s\n",
      "Downloading      ━━━━━━━━━━━━━━━━━━━━━━━ 102.3MB                            0.8s\n",
      "Extracting  (23) ━━━━━━━━╸\u001b[33m━━━━━━━━━━━━━━\u001b[0m      15 commonmark                 2.4s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 2.7s\n",
      "Downloading      ━━━━━━━━━━━━━━━━━━━━━━━ 102.3MB                            0.8s\n",
      "Extracting  (23) ━━━━━━━━╸\u001b[33m━━━━━━━━━━━━━━\u001b[0m      15 commonmark                 2.5s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 2.8s\n",
      "Downloading      ━━━━━━━━━━━━━━━━━━━━━━━ 102.3MB                            0.8s\n",
      "Extracting  (23) ━━━━━━━━╸\u001b[33m━━━━━━━━━━━━━━\u001b[0m      15 cycler                     2.6s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 2.9s\n",
      "Downloading      ━━━━━━━━━━━━━━━━━━━━━━━ 102.3MB                            0.8s\n",
      "Extracting  (23) ━━━━━━━━╸\u001b[33m━━━━━━━━━━━━━━\u001b[0m      15 cycler                     2.7s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 3.0s\n",
      "Downloading      ━━━━━━━━━━━━━━━━━━━━━━━ 102.3MB                            0.8s\n",
      "Extracting  (23) ━━━━━━━━╸\u001b[33m━━━━━━━━━━━━━━\u001b[0m      15 cycler                     2.8s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 3.1s\n",
      "Downloading      ━━━━━━━━━━━━━━━━━━━━━━━ 102.3MB                            0.8s\n",
      "Extracting  (23) ━━━━━━━━╸\u001b[33m━━━━━━━━━━━━━━\u001b[0m      15 cycler                     2.9s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 3.2s\n",
      "Downloading      ━━━━━━━━━━━━━━━━━━━━━━━ 102.3MB                            0.8s\n",
      "Extracting  (23) ━━━━━━━━╸\u001b[33m━━━━━━━━━━━━━━\u001b[0m      15 dataclasses                3.0s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 3.3s\n",
      "Downloading      ━━━━━━━━━━━━━━━━━━━━━━━ 102.3MB                            0.8s\n",
      "Extracting  (23) ━━━━━━━━╸\u001b[33m━━━━━━━━━━━━━━\u001b[0m      15 dataclasses                3.1s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 3.4s\n",
      "Downloading      ━━━━━━━━━━━━━━━━━━━━━━━ 102.3MB                            0.8s\n",
      "Extracting  (23) ━━━━━━━━╸\u001b[33m━━━━━━━━━━━━━━\u001b[0m      15 dataclasses                3.2s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 3.5s\n",
      "Downloading      ━━━━━━━━━━━━━━━━━━━━━━━ 102.3MB                            0.8s\n",
      "Extracting  (23) ━━━━━━━━╸\u001b[33m━━━━━━━━━━━━━━\u001b[0m      15 dataclasses                3.3s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 3.6s\n",
      "Downloading      ━━━━━━━━━━━━━━━━━━━━━━━ 102.3MB                            0.8s\n",
      "Extracting  (19) ━━━━━━━━━━╸\u001b[33m━━━━━━━━━━━━\u001b[0m      19 fonttools                  3.4s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 3.7s\n",
      "Downloading      ━━━━━━━━━━━━━━━━━━━━━━━ 102.3MB                            0.8s\n",
      "Extracting  (14) ━━━━━━━━━━━━━╸\u001b[33m━━━━━━━━━\u001b[0m      24 fonttools                  3.5s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 3.8s\n",
      "Downloading      ━━━━━━━━━━━━━━━━━━━━━━━ 102.3MB                            0.8s\n",
      "Extracting   (6) ━━━━━━━━━━━━━━━━━━╸\u001b[33m━━━━\u001b[0m      32 fonttools                  3.6s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 3.9s\n",
      "Downloading      ━━━━━━━━━━━━━━━━━━━━━━━ 102.3MB                            0.8s\n",
      "Extracting   (6) ━━━━━━━━━━━━━━━━━━╸\u001b[33m━━━━\u001b[0m      32 fonttools                  3.7s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 4.0s\n",
      "Downloading      ━━━━━━━━━━━━━━━━━━━━━━━ 102.3MB                            0.8s\n",
      "Extracting   (6) ━━━━━━━━━━━━━━━━━━╸\u001b[33m━━━━\u001b[0m      32 matplotlib-base            3.8s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 4.1s\n",
      "Downloading      ━━━━━━━━━━━━━━━━━━━━━━━ 102.3MB                            0.8s\n",
      "Extracting   (2) ━━━━━━━━━━━━━━━━━━━━╸\u001b[33m━━\u001b[0m      36 matplotlib-base            3.9s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 4.2s\n",
      "Downloading      ━━━━━━━━━━━━━━━━━━━━━━━ 102.3MB                            0.8s\n",
      "Extracting   (2) ━━━━━━━━━━━━━━━━━━━━╸\u001b[33m━━\u001b[0m      36 matplotlib-base            4.0s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 4.3s\n",
      "Downloading      ━━━━━━━━━━━━━━━━━━━━━━━ 102.3MB                            0.8s\n",
      "Extracting   (2) ━━━━━━━━━━━━━━━━━━━━╸\u001b[33m━━\u001b[0m      36 matplotlib-base            4.1s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 4.4s\n",
      "Downloading      ━━━━━━━━━━━━━━━━━━━━━━━ 102.3MB                            0.8s\n",
      "Extracting   (2) ━━━━━━━━━━━━━━━━━━━━╸\u001b[33m━━\u001b[0m      36 pillow                     4.2s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 4.5s\n",
      "Downloading      ━━━━━━━━━━━━━━━━━━━━━━━ 102.3MB                            0.8s\n",
      "Extracting   (2) ━━━━━━━━━━━━━━━━━━━━╸\u001b[33m━━\u001b[0m      36 pillow                     4.3s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 4.6s\n",
      "Downloading      ━━━━━━━━━━━━━━━━━━━━━━━ 102.3MB                            0.8s\n",
      "Extracting   (2) ━━━━━━━━━━━━━━━━━━━━╸\u001b[33m━━\u001b[0m      36 pillow                     4.4s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 4.7s\n",
      "Downloading      ━━━━━━━━━━━━━━━━━━━━━━━ 102.3MB                            0.8s\n",
      "Extracting   (2) ━━━━━━━━━━━━━━━━━━━━╸\u001b[33m━━\u001b[0m      36 pillow                     4.5s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 4.8s\n",
      "Downloading      ━━━━━━━━━━━━━━━━━━━━━━━ 102.3MB                            0.8s\n",
      "Extracting   (1) ━━━━━━━━━━━━━━━━━━━━━╸\u001b[33m━\u001b[0m      37 pillow                     4.6s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 4.9s\n",
      "Downloading      ━━━━━━━━━━━━━━━━━━━━━━━ 102.3MB                            0.8s\n",
      "Extracting   (1) ━━━━━━━━━━━━━━━━━━━━━╸\u001b[33m━\u001b[0m      37 pillow                     4.7s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 5.0s\n",
      "Downloading      ━━━━━━━━━━━━━━━━━━━━━━━ 102.3MB                            0.8s\n",
      "Extracting   (1) ━━━━━━━━━━━━━━━━━━━━━╸\u001b[33m━\u001b[0m      37 pillow                     4.8s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 5.1s\n",
      "Downloading      ━━━━━━━━━━━━━━━━━━━━━━━ 102.3MB                            0.8s\n",
      "Extracting   (1) ━━━━━━━━━━━━━━━━━━━━━╸\u001b[33m━\u001b[0m      37 pillow                     4.9s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 5.2s\n",
      "Downloading      ━━━━━━━━━━━━━━━━━━━━━━━ 102.3MB                            0.8s\n",
      "Extracting   (1) ━━━━━━━━━━━━━━━━━━━━━╸\u001b[33m━\u001b[0m      37 pillow                     5.0s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 5.3s\n",
      "Downloading      ━━━━━━━━━━━━━━━━━━━━━━━ 102.3MB                            0.8s\n",
      "Extracting   (1) ━━━━━━━━━━━━━━━━━━━━━╸\u001b[33m━\u001b[0m      37 pillow                     5.1s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 5.4s\n",
      "Downloading      ━━━━━━━━━━━━━━━━━━━━━━━ 102.3MB                            0.8s\n",
      "Extracting   (1) ━━━━━━━━━━━━━━━━━━━━━╸\u001b[33m━\u001b[0m      37 pillow                     5.2s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 5.5s\n",
      "Downloading      ━━━━━━━━━━━━━━━━━━━━━━━ 102.3MB                            0.8s\n",
      "Extracting   (1) ━━━━━━━━━━━━━━━━━━━━━╸\u001b[33m━\u001b[0m      37 pillow                     5.3s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 5.6s\n",
      "Downloading      ━━━━━━━━━━━━━━━━━━━━━━━ 102.3MB                            0.8s\n",
      "Extracting   (1) ━━━━━━━━━━━━━━━━━━━━━╸\u001b[33m━\u001b[0m      37 pillow                     5.4s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 5.7s\n",
      "Downloading      ━━━━━━━━━━━━━━━━━━━━━━━ 102.3MB                            0.8s\n",
      "Extracting   (1) ━━━━━━━━━━━━━━━━━━━━━╸\u001b[33m━\u001b[0m      37 pillow                     5.5s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 5.8s\n",
      "Downloading      ━━━━━━━━━━━━━━━━━━━━━━━ 102.3MB                            0.8s\n",
      "Extracting   (1) ━━━━━━━━━━━━━━━━━━━━━╸\u001b[33m━\u001b[0m      37 pillow                     5.6s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 5.9s\n",
      "Downloading      ━━━━━━━━━━━━━━━━━━━━━━━ 102.3MB                            0.8s\n",
      "Extracting   (1) ━━━━━━━━━━━━━━━━━━━━━╸\u001b[33m━\u001b[0m      37 pillow                     5.7s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 6.0s\n",
      "Downloading      ━━━━━━━━━━━━━━━━━━━━━━━ 102.3MB                            0.8s\n",
      "Extracting   (1) ━━━━━━━━━━━━━━━━━━━━━╸\u001b[33m━\u001b[0m      37 pillow                     5.8s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 6.1s\n",
      "Downloading      ━━━━━━━━━━━━━━━━━━━━━━━ 102.3MB                            0.8s\n",
      "Extracting   (1) ━━━━━━━━━━━━━━━━━━━━━╸\u001b[33m━\u001b[0m      37 pillow                     5.9s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 6.2s\n",
      "Downloading      ━━━━━━━━━━━━━━━━━━━━━━━ 102.3MB                            0.8s\n",
      "Extracting   (1) ━━━━━━━━━━━━━━━━━━━━━╸\u001b[33m━\u001b[0m      37 pillow                     6.0s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 6.3s\n",
      "Downloading      ━━━━━━━━━━━━━━━━━━━━━━━ 102.3MB                            0.8s\n",
      "Extracting   (1) ━━━━━━━━━━━━━━━━━━━━━╸\u001b[33m━\u001b[0m      37 pillow                     6.1s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 6.4s\n",
      "Downloading      ━━━━━━━━━━━━━━━━━━━━━━━ 102.3MB                            0.8s\n",
      "Extracting   (1) ━━━━━━━━━━━━━━━━━━━━━╸\u001b[33m━\u001b[0m      37 pillow                     6.2s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 6.5s\n",
      "Downloading      ━━━━━━━━━━━━━━━━━━━━━━━ 102.3MB                            0.8s\n",
      "Extracting   (1) ━━━━━━━━━━━━━━━━━━━━━╸\u001b[33m━\u001b[0m      37 pillow                     6.3s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 6.6s\n",
      "Downloading      ━━━━━━━━━━━━━━━━━━━━━━━ 102.3MB                            0.8s\n",
      "Extracting   (1) ━━━━━━━━━━━━━━━━━━━━━╸\u001b[33m━\u001b[0m      37 pillow                     6.4s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 6.7s\n",
      "Downloading      ━━━━━━━━━━━━━━━━━━━━━━━ 102.3MB                            0.8s\n",
      "Extracting   (1) ━━━━━━━━━━━━━━━━━━━━━╸\u001b[33m━\u001b[0m      37 pillow                     6.5s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 6.8s\n",
      "Downloading      ━━━━━━━━━━━━━━━━━━━━━━━ 102.3MB                            0.8s\n",
      "Extracting   (1) ━━━━━━━━━━━━━━━━━━━━━╸\u001b[33m━\u001b[0m      37 pillow                     6.6s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 6.9s\n",
      "Downloading      ━━━━━━━━━━━━━━━━━━━━━━━ 102.3MB                            0.8s\n",
      "Extracting   (1) ━━━━━━━━━━━━━━━━━━━━━╸\u001b[33m━\u001b[0m      37 pillow                     6.7s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 7.0s\n",
      "Downloading      ━━━━━━━━━━━━━━━━━━━━━━━ 102.3MB                            0.8s\n",
      "Extracting   (1) ━━━━━━━━━━━━━━━━━━━━━╸\u001b[33m━\u001b[0m      37 pillow                     6.8s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 7.1s\n",
      "Downloading      ━━━━━━━━━━━━━━━━━━━━━━━ 102.3MB                            0.8s\n",
      "Extracting   (1) ━━━━━━━━━━━━━━━━━━━━━╸\u001b[33m━\u001b[0m      37 pillow                     6.9s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 7.2s\n",
      "Downloading      ━━━━━━━━━━━━━━━━━━━━━━━ 102.3MB                            0.8s\n",
      "Extracting   (1) ━━━━━━━━━━━━━━━━━━━━━╸\u001b[33m━\u001b[0m      37 pillow                     7.0s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 7.3s\n",
      "Downloading      ━━━━━━━━━━━━━━━━━━━━━━━ 102.3MB                            0.8s\n",
      "Extracting   (1) ━━━━━━━━━━━━━━━━━━━━━╸\u001b[33m━\u001b[0m      37 pillow                     7.1s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 7.4s\n",
      "Downloading      ━━━━━━━━━━━━━━━━━━━━━━━ 102.3MB                            0.8s\n",
      "Extracting   (1) ━━━━━━━━━━━━━━━━━━━━━╸\u001b[33m━\u001b[0m      37 pillow                     7.2s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 7.5s\n",
      "Downloading      ━━━━━━━━━━━━━━━━━━━━━━━ 102.3MB                            0.8s\n",
      "Extracting   (1) ━━━━━━━━━━━━━━━━━━━━━╸\u001b[33m━\u001b[0m      37 pillow                     7.3s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G[+] 7.6s\n",
      "Downloading      ━━━━━━━━━━━━━━━━━━━━━━━ 102.3MB                            0.8s\n",
      "Extracting   (1) ━━━━━━━━━━━━━━━━━━━━━╸\u001b[33m━\u001b[0m      37 pillow                     7.4s\u001b[2K\u001b[1A\u001b[2K\u001b[1A\u001b[2K\u001b[0G\u001b[?25hPreparing transaction: done\n",
      "Verifying transaction: done\n",
      "Executing transaction: done\n"
     ]
    }
   ],
   "source": [
    "#Mambaforge\n",
    "!curl -L -O https://github.com/conda-forge/miniforge/releases/latest/download/Mambaforge-$(uname)-$(uname -m).sh\n",
    "!bash Mambaforge-$(uname)-$(uname -m).sh -b -u -p $HOME/mambaforge\n",
    "\n",
    "#Trimmomatic, Fastqc, bowtie, samtools, rsem\n",
    "!$HOME/mambaforge/bin/mamba install -y -c conda-forge -c bioconda trimmomatic fastqc bowtie samtools rsem multiqc"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ab298b90-0f0f-468e-adf3-d6c74a8f9614",
   "metadata": {},
   "source": [
    "# Download Files\n",
    "\n",
    "We need to create the folders we will use, as well as download the fastq and reference files that will be used for analysis. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "812b935a-5e24-4d4d-8a86-245a17b1fb2e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Copying gs://hon_350/example_truncated_fastq/trunc_SE6151_SA60184_S1_L004_R1_001.fastq.gz...\n",
      "Copying gs://hon_350/example_truncated_fastq/trunc_SE6151_SA60184_S1_L004_R2_001.fastq.gz...\n",
      "Copying gs://hon_350/example_truncated_fastq/trunc_SE6151_SA60185_S2_L004_R1_001.fastq.gz...\n",
      "/ [3 files][  5.4 MiB/  5.4 MiB]                                                \n",
      "==> NOTE: You are performing a sequence of gsutil operations that may\n",
      "run significantly faster if you instead use gsutil -m cp ... Please\n",
      "see the -m section under \"gsutil help options\" for further information\n",
      "about when gsutil -m can be advantageous.\n",
      "\n",
      "Copying gs://hon_350/example_truncated_fastq/trunc_SE6151_SA60185_S2_L004_R2_001.fastq.gz...\n",
      "Copying gs://hon_350/example_truncated_fastq/trunc_SE6151_SA60186_S3_L004_R1_001.fastq.gz...\n",
      "Copying gs://hon_350/example_truncated_fastq/trunc_SE6151_SA60186_S3_L004_R2_001.fastq.gz...\n",
      "Copying gs://hon_350/example_truncated_fastq/trunc_SE6151_SA60187_S4_L004_R1_001.fastq.gz...\n",
      "Copying gs://hon_350/example_truncated_fastq/trunc_SE6151_SA60187_S4_L004_R2_001.fastq.gz...\n",
      "Copying gs://hon_350/example_truncated_fastq/trunc_SE6151_SA60188_S5_L004_R1_001.fastq.gz...\n",
      "Copying gs://hon_350/example_truncated_fastq/trunc_SE6151_SA60188_S5_L004_R2_001.fastq.gz...\n",
      "Copying gs://hon_350/example_truncated_fastq/trunc_SE6151_SA60189_S6_L004_R1_001.fastq.gz...\n",
      "Copying gs://hon_350/example_truncated_fastq/trunc_SE6151_SA60189_S6_L004_R2_001.fastq.gz...\n",
      "Copying gs://hon_350/example_truncated_fastq/trunc_SE6151_SA60190_S7_L004_R1_001.fastq.gz...\n",
      "Copying gs://hon_350/example_truncated_fastq/trunc_SE6151_SA60190_S7_L004_R2_001.fastq.gz...\n",
      "Copying gs://hon_350/example_truncated_fastq/trunc_SE6151_SA60191_S8_L004_R1_001.fastq.gz...\n",
      "Copying gs://hon_350/example_truncated_fastq/trunc_SE6151_SA60191_S8_L004_R2_001.fastq.gz...\n",
      "\\ [16 files][ 26.0 MiB/ 26.0 MiB]                                               \n",
      "Operation completed over 16 objects/26.0 MiB.                                    \n",
      "Copying gs://hon_350/reference/TruSeq3-PE.fa...\n",
      "/ [1 files][   93.0 B/   93.0 B]                                                \n",
      "Operation completed over 1 objects/93.0 B.                                       \n",
      "Copying gs://hon_350/reference/rsem_grcz11.108_transcriptome.tar.gz...\n",
      "| [1 files][207.0 MiB/207.0 MiB]                                                \n",
      "Operation completed over 1 objects/207.0 MiB.                                    \n",
      "unzipping...\n",
      "data/reference/zebrafish_rsem/\n",
      "data/reference/zebrafish_rsem/GRCz11_108.4.ebwt\n",
      "data/reference/zebrafish_rsem/GRCz11_108.chrlist\n",
      "data/reference/zebrafish_rsem/GRCz11_108.2.ebwt\n",
      "data/reference/zebrafish_rsem/.ipynb_checkpoints/\n",
      "data/reference/zebrafish_rsem/GRCz11_108.n2g.idx.fa\n",
      "data/reference/zebrafish_rsem/GRCz11_108.seq\n",
      "data/reference/zebrafish_rsem/GRCz11_108.1.ebwt\n",
      "data/reference/zebrafish_rsem/GRCz11_108.rev.1.ebwt\n",
      "data/reference/zebrafish_rsem/GRCz11_108.grp\n",
      "data/reference/zebrafish_rsem/GRCz11_108.idx.fa\n",
      "data/reference/zebrafish_rsem/GRCz11_108.rev.2.ebwt\n",
      "data/reference/zebrafish_rsem/GRCz11_108.ti\n",
      "data/reference/zebrafish_rsem/GRCz11_108.transcripts.fa\n",
      "data/reference/zebrafish_rsem/GRCz11_108.3.ebwt\n",
      "finished unzipping.\n"
     ]
    }
   ],
   "source": [
    "## Create Directories to hold files\n",
    "!mkdir -p data\n",
    "!mkdir -p data/raw_fastq\n",
    "!mkdir -p data/trimmed\n",
    "!mkdir -p data/fastqc\n",
    "!mkdir -p data/multiqc\n",
    "!mkdir -p data/reference\n",
    "!mkdir -p data/rsem\n",
    "\n",
    "## Download Fastq files\n",
    "\n",
    "# Download all runs from the Google folder bucket\n",
    "!gsutil cp gs://hon_350/example_truncated_fastq/* data/raw_fastq\n",
    "\n",
    "## Adapter sequences file for trimmomatic\n",
    "!gsutil cp gs://hon_350/reference/TruSeq3-PE.fa data/reference/TruSeq3-PE.fa\n",
    "\n",
    "## (Optional) Download reference files for building transcriptome index\n",
    "#!gsutil cp gs://hon_350/referebce/GRCz11.dna_sm.toplevel.fa.gz data/reference/GRCz11.dna_sm.toplevel.fa.gz\n",
    "#!gsutil cp gs://hon_350/referebce/GRCz11.108.gtf.gz data/reference/GRCz11.108.gtf.gz\n",
    "#!echo 'unzipping...'\n",
    "#!gzip -d data/reference/GRCz11.dna_sm.toplevel.fa.gz\n",
    "#!gzip -d data/reference/GRCz11.107.gtf.gz\n",
    "#!echo 'finished unzipping.'\n",
    "\n",
    "## Download and unzip premade index files\n",
    "!gsutil cp gs://hon_350/reference/rsem_grcz11.108_transcriptome.tar.gz data/reference/rsem_grcz11.108_transcriptome.tar.gz\n",
    "!echo 'unzipping...'\n",
    "!tar -xzvf data/reference/rsem_grcz11.108_transcriptome.tar.gz\n",
    "!rm data/reference/rsem_grcz11.108_transcriptome.tar.gz\n",
    "!echo 'finished unzipping.'"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c0767b47-81e0-4b68-825f-ee1795b3c6e8",
   "metadata": {},
   "source": [
    "# Trimmomatic\n",
    "\n",
    "Trimmomatic is used to trim fastq files to remove adapters, and low quality score bases in reads.\n",
    "\n",
    "A nice guide on trimmomatic parameters: https://datacarpentry.org/wrangling-genomics/03-trimming/index.html \n",
    "\n",
    "Trimmomatic works on one paired or single end run at a time.\n",
    "\n",
    "However we have dozens of paired-end reads. \n",
    "\n",
    "To speed things up, this example 'pipes in' a list of run ids into trimmomatic to run sequentially."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "979ae9a4-8a2d-46d2-8038-38d16f85521b",
   "metadata": {
    "scrolled": true,
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "trunc_SE6151_SA60184_S1_L004_R\n",
      "trunc_SE6151_SA60185_S2_L004_R\n",
      "trunc_SE6151_SA60186_S3_L004_R\n",
      "trunc_SE6151_SA60187_S4_L004_R\n",
      "trunc_SE6151_SA60188_S5_L004_R\n",
      "trunc_SE6151_SA60189_S6_L004_R\n",
      "trunc_SE6151_SA60190_S7_L004_R\n",
      "trunc_SE6151_SA60191_S8_L004_R\n",
      " \n",
      "TrimmomaticPE: Started with arguments:\n",
      " -threads 2 data/raw_fastq/trunc_SE6151_SA60184_S1_L004_R1_001.fastq.gz data/raw_fastq/trunc_SE6151_SA60184_S1_L004_R2_001.fastq.gz data/trimmed/trunc_SE6151_SA60184_S1_L004_R1_trimmed.fastq.gz data/trimmed/trunc_SE6151_SA60184_S1_L004_R1_orphans.fastq.gz data/trimmed/trunc_SE6151_SA60184_S1_L004_R2_trimmed.fastq.gz data/trimmed/trunc_SE6151_SA60184_S1_L004_R2_orphans.fastq.gz ILLUMINACLIP:data/reference/TruSeq3-PE.fa:2:30:10:2:keepBothReads LEADING:3 TRAILING:3 MINLEN:36\n",
      "Using PrefixPair: 'TACACTCTTTCCCTACACGACGCTCTTCCGATCT' and 'GTGACTGGAGTTCAGACGTGTGCTCTTCCGATCT'\n",
      "ILLUMINACLIP: Using 1 prefix pairs, 0 forward/reverse sequences, 0 forward only sequences, 0 reverse only sequences\n",
      "Quality encoding detected as phred33\n",
      "Input Read Pairs: 25000 Both Surviving: 21329 (85.32%) Forward Only Surviving: 3669 (14.68%) Reverse Only Surviving: 0 (0.00%) Dropped: 2 (0.01%)\n",
      "TrimmomaticPE: Completed successfully\n",
      "TrimmomaticPE: Started with arguments:\n",
      " -threads 2 data/raw_fastq/trunc_SE6151_SA60185_S2_L004_R1_001.fastq.gz data/raw_fastq/trunc_SE6151_SA60185_S2_L004_R2_001.fastq.gz data/trimmed/trunc_SE6151_SA60185_S2_L004_R1_trimmed.fastq.gz data/trimmed/trunc_SE6151_SA60185_S2_L004_R1_orphans.fastq.gz data/trimmed/trunc_SE6151_SA60185_S2_L004_R2_trimmed.fastq.gz data/trimmed/trunc_SE6151_SA60185_S2_L004_R2_orphans.fastq.gz ILLUMINACLIP:data/reference/TruSeq3-PE.fa:2:30:10:2:keepBothReads LEADING:3 TRAILING:3 MINLEN:36\n",
      "Using PrefixPair: 'TACACTCTTTCCCTACACGACGCTCTTCCGATCT' and 'GTGACTGGAGTTCAGACGTGTGCTCTTCCGATCT'\n",
      "ILLUMINACLIP: Using 1 prefix pairs, 0 forward/reverse sequences, 0 forward only sequences, 0 reverse only sequences\n",
      "Quality encoding detected as phred33\n",
      "Input Read Pairs: 25000 Both Surviving: 19635 (78.54%) Forward Only Surviving: 5361 (21.44%) Reverse Only Surviving: 0 (0.00%) Dropped: 4 (0.02%)\n",
      "TrimmomaticPE: Completed successfully\n",
      "TrimmomaticPE: Started with arguments:\n",
      " -threads 2 data/raw_fastq/trunc_SE6151_SA60186_S3_L004_R1_001.fastq.gz data/raw_fastq/trunc_SE6151_SA60186_S3_L004_R2_001.fastq.gz data/trimmed/trunc_SE6151_SA60186_S3_L004_R1_trimmed.fastq.gz data/trimmed/trunc_SE6151_SA60186_S3_L004_R1_orphans.fastq.gz data/trimmed/trunc_SE6151_SA60186_S3_L004_R2_trimmed.fastq.gz data/trimmed/trunc_SE6151_SA60186_S3_L004_R2_orphans.fastq.gz ILLUMINACLIP:data/reference/TruSeq3-PE.fa:2:30:10:2:keepBothReads LEADING:3 TRAILING:3 MINLEN:36\n",
      "Using PrefixPair: 'TACACTCTTTCCCTACACGACGCTCTTCCGATCT' and 'GTGACTGGAGTTCAGACGTGTGCTCTTCCGATCT'\n",
      "ILLUMINACLIP: Using 1 prefix pairs, 0 forward/reverse sequences, 0 forward only sequences, 0 reverse only sequences\n",
      "Quality encoding detected as phred33\n",
      "Input Read Pairs: 25000 Both Surviving: 17563 (70.25%) Forward Only Surviving: 7432 (29.73%) Reverse Only Surviving: 0 (0.00%) Dropped: 5 (0.02%)\n",
      "TrimmomaticPE: Completed successfully\n",
      "TrimmomaticPE: Started with arguments:\n",
      " -threads 2 data/raw_fastq/trunc_SE6151_SA60187_S4_L004_R1_001.fastq.gz data/raw_fastq/trunc_SE6151_SA60187_S4_L004_R2_001.fastq.gz data/trimmed/trunc_SE6151_SA60187_S4_L004_R1_trimmed.fastq.gz data/trimmed/trunc_SE6151_SA60187_S4_L004_R1_orphans.fastq.gz data/trimmed/trunc_SE6151_SA60187_S4_L004_R2_trimmed.fastq.gz data/trimmed/trunc_SE6151_SA60187_S4_L004_R2_orphans.fastq.gz ILLUMINACLIP:data/reference/TruSeq3-PE.fa:2:30:10:2:keepBothReads LEADING:3 TRAILING:3 MINLEN:36\n",
      "Using PrefixPair: 'TACACTCTTTCCCTACACGACGCTCTTCCGATCT' and 'GTGACTGGAGTTCAGACGTGTGCTCTTCCGATCT'\n",
      "ILLUMINACLIP: Using 1 prefix pairs, 0 forward/reverse sequences, 0 forward only sequences, 0 reverse only sequences\n",
      "Quality encoding detected as phred33\n",
      "Input Read Pairs: 25000 Both Surviving: 21689 (86.76%) Forward Only Surviving: 3304 (13.22%) Reverse Only Surviving: 0 (0.00%) Dropped: 7 (0.03%)\n",
      "TrimmomaticPE: Completed successfully\n",
      "TrimmomaticPE: Started with arguments:\n",
      " -threads 2 data/raw_fastq/trunc_SE6151_SA60188_S5_L004_R1_001.fastq.gz data/raw_fastq/trunc_SE6151_SA60188_S5_L004_R2_001.fastq.gz data/trimmed/trunc_SE6151_SA60188_S5_L004_R1_trimmed.fastq.gz data/trimmed/trunc_SE6151_SA60188_S5_L004_R1_orphans.fastq.gz data/trimmed/trunc_SE6151_SA60188_S5_L004_R2_trimmed.fastq.gz data/trimmed/trunc_SE6151_SA60188_S5_L004_R2_orphans.fastq.gz ILLUMINACLIP:data/reference/TruSeq3-PE.fa:2:30:10:2:keepBothReads LEADING:3 TRAILING:3 MINLEN:36\n",
      "Using PrefixPair: 'TACACTCTTTCCCTACACGACGCTCTTCCGATCT' and 'GTGACTGGAGTTCAGACGTGTGCTCTTCCGATCT'\n",
      "ILLUMINACLIP: Using 1 prefix pairs, 0 forward/reverse sequences, 0 forward only sequences, 0 reverse only sequences\n",
      "Quality encoding detected as phred33\n",
      "Input Read Pairs: 25000 Both Surviving: 24736 (98.94%) Forward Only Surviving: 264 (1.06%) Reverse Only Surviving: 0 (0.00%) Dropped: 0 (0.00%)\n",
      "TrimmomaticPE: Completed successfully\n",
      "TrimmomaticPE: Started with arguments:\n",
      " -threads 2 data/raw_fastq/trunc_SE6151_SA60189_S6_L004_R1_001.fastq.gz data/raw_fastq/trunc_SE6151_SA60189_S6_L004_R2_001.fastq.gz data/trimmed/trunc_SE6151_SA60189_S6_L004_R1_trimmed.fastq.gz data/trimmed/trunc_SE6151_SA60189_S6_L004_R1_orphans.fastq.gz data/trimmed/trunc_SE6151_SA60189_S6_L004_R2_trimmed.fastq.gz data/trimmed/trunc_SE6151_SA60189_S6_L004_R2_orphans.fastq.gz ILLUMINACLIP:data/reference/TruSeq3-PE.fa:2:30:10:2:keepBothReads LEADING:3 TRAILING:3 MINLEN:36\n",
      "Using PrefixPair: 'TACACTCTTTCCCTACACGACGCTCTTCCGATCT' and 'GTGACTGGAGTTCAGACGTGTGCTCTTCCGATCT'\n",
      "ILLUMINACLIP: Using 1 prefix pairs, 0 forward/reverse sequences, 0 forward only sequences, 0 reverse only sequences\n",
      "Quality encoding detected as phred33\n",
      "Input Read Pairs: 25000 Both Surviving: 20582 (82.33%) Forward Only Surviving: 4400 (17.60%) Reverse Only Surviving: 0 (0.00%) Dropped: 18 (0.07%)\n",
      "TrimmomaticPE: Completed successfully\n",
      "TrimmomaticPE: Started with arguments:\n",
      " -threads 2 data/raw_fastq/trunc_SE6151_SA60190_S7_L004_R1_001.fastq.gz data/raw_fastq/trunc_SE6151_SA60190_S7_L004_R2_001.fastq.gz data/trimmed/trunc_SE6151_SA60190_S7_L004_R1_trimmed.fastq.gz data/trimmed/trunc_SE6151_SA60190_S7_L004_R1_orphans.fastq.gz data/trimmed/trunc_SE6151_SA60190_S7_L004_R2_trimmed.fastq.gz data/trimmed/trunc_SE6151_SA60190_S7_L004_R2_orphans.fastq.gz ILLUMINACLIP:data/reference/TruSeq3-PE.fa:2:30:10:2:keepBothReads LEADING:3 TRAILING:3 MINLEN:36\n",
      "Using PrefixPair: 'TACACTCTTTCCCTACACGACGCTCTTCCGATCT' and 'GTGACTGGAGTTCAGACGTGTGCTCTTCCGATCT'\n",
      "ILLUMINACLIP: Using 1 prefix pairs, 0 forward/reverse sequences, 0 forward only sequences, 0 reverse only sequences\n",
      "Quality encoding detected as phred33\n",
      "Input Read Pairs: 25000 Both Surviving: 3281 (13.12%) Forward Only Surviving: 21719 (86.88%) Reverse Only Surviving: 0 (0.00%) Dropped: 0 (0.00%)\n",
      "TrimmomaticPE: Completed successfully\n",
      "TrimmomaticPE: Started with arguments:\n",
      " -threads 2 data/raw_fastq/trunc_SE6151_SA60191_S8_L004_R1_001.fastq.gz data/raw_fastq/trunc_SE6151_SA60191_S8_L004_R2_001.fastq.gz data/trimmed/trunc_SE6151_SA60191_S8_L004_R1_trimmed.fastq.gz data/trimmed/trunc_SE6151_SA60191_S8_L004_R1_orphans.fastq.gz data/trimmed/trunc_SE6151_SA60191_S8_L004_R2_trimmed.fastq.gz data/trimmed/trunc_SE6151_SA60191_S8_L004_R2_orphans.fastq.gz ILLUMINACLIP:data/reference/TruSeq3-PE.fa:2:30:10:2:keepBothReads LEADING:3 TRAILING:3 MINLEN:36\n",
      "Using PrefixPair: 'TACACTCTTTCCCTACACGACGCTCTTCCGATCT' and 'GTGACTGGAGTTCAGACGTGTGCTCTTCCGATCT'\n",
      "ILLUMINACLIP: Using 1 prefix pairs, 0 forward/reverse sequences, 0 forward only sequences, 0 reverse only sequences\n",
      "Quality encoding detected as phred33\n",
      "Input Read Pairs: 25000 Both Surviving: 2328 (9.31%) Forward Only Surviving: 22672 (90.69%) Reverse Only Surviving: 0 (0.00%) Dropped: 0 (0.00%)\n",
      "TrimmomaticPE: Completed successfully\n"
     ]
    }
   ],
   "source": [
    "# Run trimmomatic\n",
    "# trimmomatic works on one paired-end reads file\n",
    "# at a time. an example might look like this:\n",
    "\n",
    "#!trimmomatic PE -threads 16 data/raw_fastq/trunc_SE6151_SA60191_S8_L004_R1_001.fastq.gz data/raw_fastq/trunc_SE6151_SA60191_S8_L004_R2_001.fastq.gz data/trimmed/SA60191_1_trimmed.fastq.gz data/trimmed/SA60191_1_orphans.fastq.gz data/trimmed/SA60191_2_trimmed.fastq.gz data/trimmed/SA60191_2_orphans.fastq.gz ILLUMINACLIP:data/reference/TruSeq3-PE.fa:2:30:10:2:keepBothReads LEADING:3 TRAILING:3 MINLEN:36\n",
    "\n",
    "# for the sake of this example a list is \n",
    "# created using filenames in the fastq\n",
    "# directory, and those names are piped in\n",
    "# iteratively to the command function\n",
    "\n",
    "# a list of the runs is made\n",
    "!ls data/raw_fastq/*R1* |  sed 's|data/raw_fastq/||g' | sed 's|1_001.fastq.gz||g' > run_ids.txt\n",
    "\n",
    "#what the list looks like\n",
    "!cat run_ids.txt\n",
    "!echo ' '\n",
    "\n",
    "#the list is piped into trimmomatic\n",
    "!cat run_ids.txt |  xargs -I {} trimmomatic PE -threads 16 data/raw_fastq/{}1_001.fastq.gz data/raw_fastq/{}2_001.fastq.gz data/trimmed/{}1_trimmed.fastq.gz data/trimmed/{}1_orphans.fastq.gz data/trimmed/{}2_trimmed.fastq.gz data/trimmed/{}2_orphans.fastq.gz ILLUMINACLIP:data/reference/TruSeq3-PE.fa:2:30:10:2:keepBothReads LEADING:3 TRAILING:3 MINLEN:36\n",
    "\n",
    "#may not be necessary,\n",
    "#but just to save space, delete raw fastq files after\n",
    "!rm -r data/raw_fastq"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6bbf0b43-8d1c-4bfb-9d83-10312a474b51",
   "metadata": {},
   "source": [
    "# Fastqc\n",
    "\n",
    "Fastqc is a tool to inspect the quality of reads in fastq files."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "c59d25e6-3365-41ea-9b1f-3fc913661d23",
   "metadata": {
    "scrolled": true,
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Started analysis of trunc_SE6151_SA60184_S1_L004_R1_trimmed.fastq.gz\n",
      "Approx 5% complete for trunc_SE6151_SA60184_S1_L004_R1_trimmed.fastq.gz\n",
      "Approx 10% complete for trunc_SE6151_SA60184_S1_L004_R1_trimmed.fastq.gz\n",
      "Approx 15% complete for trunc_SE6151_SA60184_S1_L004_R1_trimmed.fastq.gz\n",
      "Approx 20% complete for trunc_SE6151_SA60184_S1_L004_R1_trimmed.fastq.gz\n",
      "Approx 25% complete for trunc_SE6151_SA60184_S1_L004_R1_trimmed.fastq.gz\n",
      "Approx 30% complete for trunc_SE6151_SA60184_S1_L004_R1_trimmed.fastq.gz\n",
      "Approx 35% complete for trunc_SE6151_SA60184_S1_L004_R1_trimmed.fastq.gz\n",
      "Approx 40% complete for trunc_SE6151_SA60184_S1_L004_R1_trimmed.fastq.gz\n",
      "Approx 45% complete for trunc_SE6151_SA60184_S1_L004_R1_trimmed.fastq.gz\n",
      "Approx 50% complete for trunc_SE6151_SA60184_S1_L004_R1_trimmed.fastq.gz\n",
      "Approx 55% complete for trunc_SE6151_SA60184_S1_L004_R1_trimmed.fastq.gz\n",
      "Approx 60% complete for trunc_SE6151_SA60184_S1_L004_R1_trimmed.fastq.gz\n",
      "Approx 65% complete for trunc_SE6151_SA60184_S1_L004_R1_trimmed.fastq.gz\n",
      "Approx 70% complete for trunc_SE6151_SA60184_S1_L004_R1_trimmed.fastq.gz\n",
      "Approx 75% complete for trunc_SE6151_SA60184_S1_L004_R1_trimmed.fastq.gz\n",
      "Approx 80% complete for trunc_SE6151_SA60184_S1_L004_R1_trimmed.fastq.gz\n",
      "Approx 85% complete for trunc_SE6151_SA60184_S1_L004_R1_trimmed.fastq.gz\n",
      "Approx 90% complete for trunc_SE6151_SA60184_S1_L004_R1_trimmed.fastq.gz\n",
      "Approx 95% complete for trunc_SE6151_SA60184_S1_L004_R1_trimmed.fastq.gz\n",
      "Analysis complete for trunc_SE6151_SA60184_S1_L004_R1_trimmed.fastq.gz\n",
      "Started analysis of trunc_SE6151_SA60184_S1_L004_R2_trimmed.fastq.gz\n",
      "Approx 5% complete for trunc_SE6151_SA60184_S1_L004_R2_trimmed.fastq.gz\n",
      "Approx 10% complete for trunc_SE6151_SA60184_S1_L004_R2_trimmed.fastq.gz\n",
      "Approx 15% complete for trunc_SE6151_SA60184_S1_L004_R2_trimmed.fastq.gz\n",
      "Approx 20% complete for trunc_SE6151_SA60184_S1_L004_R2_trimmed.fastq.gz\n",
      "Approx 25% complete for trunc_SE6151_SA60184_S1_L004_R2_trimmed.fastq.gz\n",
      "Approx 30% complete for trunc_SE6151_SA60184_S1_L004_R2_trimmed.fastq.gz\n",
      "Approx 35% complete for trunc_SE6151_SA60184_S1_L004_R2_trimmed.fastq.gz\n",
      "Approx 40% complete for trunc_SE6151_SA60184_S1_L004_R2_trimmed.fastq.gz\n",
      "Approx 45% complete for trunc_SE6151_SA60184_S1_L004_R2_trimmed.fastq.gz\n",
      "Approx 50% complete for trunc_SE6151_SA60184_S1_L004_R2_trimmed.fastq.gz\n",
      "Approx 55% complete for trunc_SE6151_SA60184_S1_L004_R2_trimmed.fastq.gz\n",
      "Approx 60% complete for trunc_SE6151_SA60184_S1_L004_R2_trimmed.fastq.gz\n",
      "Approx 65% complete for trunc_SE6151_SA60184_S1_L004_R2_trimmed.fastq.gz\n",
      "Approx 70% complete for trunc_SE6151_SA60184_S1_L004_R2_trimmed.fastq.gz\n",
      "Approx 75% complete for trunc_SE6151_SA60184_S1_L004_R2_trimmed.fastq.gz\n",
      "Approx 80% complete for trunc_SE6151_SA60184_S1_L004_R2_trimmed.fastq.gz\n",
      "Approx 85% complete for trunc_SE6151_SA60184_S1_L004_R2_trimmed.fastq.gz\n",
      "Approx 90% complete for trunc_SE6151_SA60184_S1_L004_R2_trimmed.fastq.gz\n",
      "Approx 95% complete for trunc_SE6151_SA60184_S1_L004_R2_trimmed.fastq.gz\n",
      "Analysis complete for trunc_SE6151_SA60184_S1_L004_R2_trimmed.fastq.gz\n",
      "Started analysis of trunc_SE6151_SA60185_S2_L004_R1_trimmed.fastq.gz\n",
      "Approx 5% complete for trunc_SE6151_SA60185_S2_L004_R1_trimmed.fastq.gz\n",
      "Approx 10% complete for trunc_SE6151_SA60185_S2_L004_R1_trimmed.fastq.gz\n",
      "Approx 15% complete for trunc_SE6151_SA60185_S2_L004_R1_trimmed.fastq.gz\n",
      "Approx 20% complete for trunc_SE6151_SA60185_S2_L004_R1_trimmed.fastq.gz\n",
      "Approx 25% complete for trunc_SE6151_SA60185_S2_L004_R1_trimmed.fastq.gz\n",
      "Approx 30% complete for trunc_SE6151_SA60185_S2_L004_R1_trimmed.fastq.gz\n",
      "Approx 35% complete for trunc_SE6151_SA60185_S2_L004_R1_trimmed.fastq.gz\n",
      "Approx 40% complete for trunc_SE6151_SA60185_S2_L004_R1_trimmed.fastq.gz\n",
      "Approx 45% complete for trunc_SE6151_SA60185_S2_L004_R1_trimmed.fastq.gz\n",
      "Approx 50% complete for trunc_SE6151_SA60185_S2_L004_R1_trimmed.fastq.gz\n",
      "Approx 55% complete for trunc_SE6151_SA60185_S2_L004_R1_trimmed.fastq.gz\n",
      "Approx 60% complete for trunc_SE6151_SA60185_S2_L004_R1_trimmed.fastq.gz\n",
      "Approx 65% complete for trunc_SE6151_SA60185_S2_L004_R1_trimmed.fastq.gz\n",
      "Approx 70% complete for trunc_SE6151_SA60185_S2_L004_R1_trimmed.fastq.gz\n",
      "Approx 75% complete for trunc_SE6151_SA60185_S2_L004_R1_trimmed.fastq.gz\n",
      "Approx 80% complete for trunc_SE6151_SA60185_S2_L004_R1_trimmed.fastq.gz\n",
      "Approx 85% complete for trunc_SE6151_SA60185_S2_L004_R1_trimmed.fastq.gz\n",
      "Approx 90% complete for trunc_SE6151_SA60185_S2_L004_R1_trimmed.fastq.gz\n",
      "Approx 95% complete for trunc_SE6151_SA60185_S2_L004_R1_trimmed.fastq.gz\n",
      "Analysis complete for trunc_SE6151_SA60185_S2_L004_R1_trimmed.fastq.gz\n",
      "Started analysis of trunc_SE6151_SA60185_S2_L004_R2_trimmed.fastq.gz\n",
      "Approx 5% complete for trunc_SE6151_SA60185_S2_L004_R2_trimmed.fastq.gz\n",
      "Approx 10% complete for trunc_SE6151_SA60185_S2_L004_R2_trimmed.fastq.gz\n",
      "Approx 15% complete for trunc_SE6151_SA60185_S2_L004_R2_trimmed.fastq.gz\n",
      "Approx 20% complete for trunc_SE6151_SA60185_S2_L004_R2_trimmed.fastq.gz\n",
      "Approx 25% complete for trunc_SE6151_SA60185_S2_L004_R2_trimmed.fastq.gz\n",
      "Approx 30% complete for trunc_SE6151_SA60185_S2_L004_R2_trimmed.fastq.gz\n",
      "Approx 35% complete for trunc_SE6151_SA60185_S2_L004_R2_trimmed.fastq.gz\n",
      "Approx 40% complete for trunc_SE6151_SA60185_S2_L004_R2_trimmed.fastq.gz\n",
      "Approx 45% complete for trunc_SE6151_SA60185_S2_L004_R2_trimmed.fastq.gz\n",
      "Approx 50% complete for trunc_SE6151_SA60185_S2_L004_R2_trimmed.fastq.gz\n",
      "Approx 55% complete for trunc_SE6151_SA60185_S2_L004_R2_trimmed.fastq.gz\n",
      "Approx 60% complete for trunc_SE6151_SA60185_S2_L004_R2_trimmed.fastq.gz\n",
      "Approx 65% complete for trunc_SE6151_SA60185_S2_L004_R2_trimmed.fastq.gz\n",
      "Approx 70% complete for trunc_SE6151_SA60185_S2_L004_R2_trimmed.fastq.gz\n",
      "Approx 75% complete for trunc_SE6151_SA60185_S2_L004_R2_trimmed.fastq.gz\n",
      "Approx 80% complete for trunc_SE6151_SA60185_S2_L004_R2_trimmed.fastq.gz\n",
      "Approx 85% complete for trunc_SE6151_SA60185_S2_L004_R2_trimmed.fastq.gz\n",
      "Approx 90% complete for trunc_SE6151_SA60185_S2_L004_R2_trimmed.fastq.gz\n",
      "Approx 95% complete for trunc_SE6151_SA60185_S2_L004_R2_trimmed.fastq.gz\n",
      "Analysis complete for trunc_SE6151_SA60185_S2_L004_R2_trimmed.fastq.gz\n",
      "Started analysis of trunc_SE6151_SA60186_S3_L004_R1_trimmed.fastq.gz\n",
      "Approx 5% complete for trunc_SE6151_SA60186_S3_L004_R1_trimmed.fastq.gz\n",
      "Approx 10% complete for trunc_SE6151_SA60186_S3_L004_R1_trimmed.fastq.gz\n",
      "Approx 15% complete for trunc_SE6151_SA60186_S3_L004_R1_trimmed.fastq.gz\n",
      "Approx 20% complete for trunc_SE6151_SA60186_S3_L004_R1_trimmed.fastq.gz\n",
      "Approx 30% complete for trunc_SE6151_SA60186_S3_L004_R1_trimmed.fastq.gz\n",
      "Approx 40% complete for trunc_SE6151_SA60186_S3_L004_R1_trimmed.fastq.gz\n",
      "Approx 45% complete for trunc_SE6151_SA60186_S3_L004_R1_trimmed.fastq.gz\n",
      "Approx 50% complete for trunc_SE6151_SA60186_S3_L004_R1_trimmed.fastq.gz\n",
      "Approx 55% complete for trunc_SE6151_SA60186_S3_L004_R1_trimmed.fastq.gz\n",
      "Approx 60% complete for trunc_SE6151_SA60186_S3_L004_R1_trimmed.fastq.gz\n",
      "Approx 65% complete for trunc_SE6151_SA60186_S3_L004_R1_trimmed.fastq.gz\n",
      "Approx 70% complete for trunc_SE6151_SA60186_S3_L004_R1_trimmed.fastq.gz\n",
      "Approx 80% complete for trunc_SE6151_SA60186_S3_L004_R1_trimmed.fastq.gz\n",
      "Approx 85% complete for trunc_SE6151_SA60186_S3_L004_R1_trimmed.fastq.gz\n",
      "Approx 90% complete for trunc_SE6151_SA60186_S3_L004_R1_trimmed.fastq.gz\n",
      "Approx 95% complete for trunc_SE6151_SA60186_S3_L004_R1_trimmed.fastq.gz\n",
      "Analysis complete for trunc_SE6151_SA60186_S3_L004_R1_trimmed.fastq.gz\n",
      "Started analysis of trunc_SE6151_SA60186_S3_L004_R2_trimmed.fastq.gz\n",
      "Approx 5% complete for trunc_SE6151_SA60186_S3_L004_R2_trimmed.fastq.gz\n",
      "Approx 10% complete for trunc_SE6151_SA60186_S3_L004_R2_trimmed.fastq.gz\n",
      "Approx 15% complete for trunc_SE6151_SA60186_S3_L004_R2_trimmed.fastq.gz\n",
      "Approx 20% complete for trunc_SE6151_SA60186_S3_L004_R2_trimmed.fastq.gz\n",
      "Approx 25% complete for trunc_SE6151_SA60186_S3_L004_R2_trimmed.fastq.gz\n",
      "Approx 30% complete for trunc_SE6151_SA60186_S3_L004_R2_trimmed.fastq.gz\n",
      "Approx 40% complete for trunc_SE6151_SA60186_S3_L004_R2_trimmed.fastq.gz\n",
      "Approx 45% complete for trunc_SE6151_SA60186_S3_L004_R2_trimmed.fastq.gz\n",
      "Approx 50% complete for trunc_SE6151_SA60186_S3_L004_R2_trimmed.fastq.gz\n",
      "Approx 55% complete for trunc_SE6151_SA60186_S3_L004_R2_trimmed.fastq.gz\n",
      "Approx 60% complete for trunc_SE6151_SA60186_S3_L004_R2_trimmed.fastq.gz\n",
      "Approx 65% complete for trunc_SE6151_SA60186_S3_L004_R2_trimmed.fastq.gz\n",
      "Approx 70% complete for trunc_SE6151_SA60186_S3_L004_R2_trimmed.fastq.gz\n",
      "Approx 80% complete for trunc_SE6151_SA60186_S3_L004_R2_trimmed.fastq.gz\n",
      "Approx 85% complete for trunc_SE6151_SA60186_S3_L004_R2_trimmed.fastq.gz\n",
      "Approx 90% complete for trunc_SE6151_SA60186_S3_L004_R2_trimmed.fastq.gz\n",
      "Approx 95% complete for trunc_SE6151_SA60186_S3_L004_R2_trimmed.fastq.gz\n",
      "Analysis complete for trunc_SE6151_SA60186_S3_L004_R2_trimmed.fastq.gz\n",
      "Started analysis of trunc_SE6151_SA60187_S4_L004_R1_trimmed.fastq.gz\n",
      "Approx 5% complete for trunc_SE6151_SA60187_S4_L004_R1_trimmed.fastq.gz\n",
      "Approx 10% complete for trunc_SE6151_SA60187_S4_L004_R1_trimmed.fastq.gz\n",
      "Approx 15% complete for trunc_SE6151_SA60187_S4_L004_R1_trimmed.fastq.gz\n",
      "Approx 20% complete for trunc_SE6151_SA60187_S4_L004_R1_trimmed.fastq.gz\n",
      "Approx 25% complete for trunc_SE6151_SA60187_S4_L004_R1_trimmed.fastq.gz\n",
      "Approx 30% complete for trunc_SE6151_SA60187_S4_L004_R1_trimmed.fastq.gz\n",
      "Approx 35% complete for trunc_SE6151_SA60187_S4_L004_R1_trimmed.fastq.gz\n",
      "Approx 40% complete for trunc_SE6151_SA60187_S4_L004_R1_trimmed.fastq.gz\n",
      "Approx 45% complete for trunc_SE6151_SA60187_S4_L004_R1_trimmed.fastq.gz\n",
      "Approx 50% complete for trunc_SE6151_SA60187_S4_L004_R1_trimmed.fastq.gz\n",
      "Approx 55% complete for trunc_SE6151_SA60187_S4_L004_R1_trimmed.fastq.gz\n",
      "Approx 60% complete for trunc_SE6151_SA60187_S4_L004_R1_trimmed.fastq.gz\n",
      "Approx 65% complete for trunc_SE6151_SA60187_S4_L004_R1_trimmed.fastq.gz\n",
      "Approx 70% complete for trunc_SE6151_SA60187_S4_L004_R1_trimmed.fastq.gz\n",
      "Approx 75% complete for trunc_SE6151_SA60187_S4_L004_R1_trimmed.fastq.gz\n",
      "Approx 80% complete for trunc_SE6151_SA60187_S4_L004_R1_trimmed.fastq.gz\n",
      "Approx 85% complete for trunc_SE6151_SA60187_S4_L004_R1_trimmed.fastq.gz\n",
      "Approx 90% complete for trunc_SE6151_SA60187_S4_L004_R1_trimmed.fastq.gz\n",
      "Approx 95% complete for trunc_SE6151_SA60187_S4_L004_R1_trimmed.fastq.gz\n",
      "Analysis complete for trunc_SE6151_SA60187_S4_L004_R1_trimmed.fastq.gz\n",
      "Started analysis of trunc_SE6151_SA60187_S4_L004_R2_trimmed.fastq.gz\n",
      "Approx 5% complete for trunc_SE6151_SA60187_S4_L004_R2_trimmed.fastq.gz\n",
      "Approx 10% complete for trunc_SE6151_SA60187_S4_L004_R2_trimmed.fastq.gz\n",
      "Approx 15% complete for trunc_SE6151_SA60187_S4_L004_R2_trimmed.fastq.gz\n",
      "Approx 20% complete for trunc_SE6151_SA60187_S4_L004_R2_trimmed.fastq.gz\n",
      "Approx 25% complete for trunc_SE6151_SA60187_S4_L004_R2_trimmed.fastq.gz\n",
      "Approx 30% complete for trunc_SE6151_SA60187_S4_L004_R2_trimmed.fastq.gz\n",
      "Approx 35% complete for trunc_SE6151_SA60187_S4_L004_R2_trimmed.fastq.gz\n",
      "Approx 40% complete for trunc_SE6151_SA60187_S4_L004_R2_trimmed.fastq.gz\n",
      "Approx 45% complete for trunc_SE6151_SA60187_S4_L004_R2_trimmed.fastq.gz\n",
      "Approx 50% complete for trunc_SE6151_SA60187_S4_L004_R2_trimmed.fastq.gz\n",
      "Approx 55% complete for trunc_SE6151_SA60187_S4_L004_R2_trimmed.fastq.gz\n",
      "Approx 60% complete for trunc_SE6151_SA60187_S4_L004_R2_trimmed.fastq.gz\n",
      "Approx 65% complete for trunc_SE6151_SA60187_S4_L004_R2_trimmed.fastq.gz\n",
      "Approx 70% complete for trunc_SE6151_SA60187_S4_L004_R2_trimmed.fastq.gz\n",
      "Approx 75% complete for trunc_SE6151_SA60187_S4_L004_R2_trimmed.fastq.gz\n",
      "Approx 80% complete for trunc_SE6151_SA60187_S4_L004_R2_trimmed.fastq.gz\n",
      "Approx 85% complete for trunc_SE6151_SA60187_S4_L004_R2_trimmed.fastq.gz\n",
      "Approx 90% complete for trunc_SE6151_SA60187_S4_L004_R2_trimmed.fastq.gz\n",
      "Approx 95% complete for trunc_SE6151_SA60187_S4_L004_R2_trimmed.fastq.gz\n",
      "Analysis complete for trunc_SE6151_SA60187_S4_L004_R2_trimmed.fastq.gz\n",
      "Started analysis of trunc_SE6151_SA60188_S5_L004_R1_trimmed.fastq.gz\n",
      "Approx 5% complete for trunc_SE6151_SA60188_S5_L004_R1_trimmed.fastq.gz\n",
      "Approx 10% complete for trunc_SE6151_SA60188_S5_L004_R1_trimmed.fastq.gz\n",
      "Approx 15% complete for trunc_SE6151_SA60188_S5_L004_R1_trimmed.fastq.gz\n",
      "Approx 20% complete for trunc_SE6151_SA60188_S5_L004_R1_trimmed.fastq.gz\n",
      "Approx 25% complete for trunc_SE6151_SA60188_S5_L004_R1_trimmed.fastq.gz\n",
      "Approx 30% complete for trunc_SE6151_SA60188_S5_L004_R1_trimmed.fastq.gz\n",
      "Approx 35% complete for trunc_SE6151_SA60188_S5_L004_R1_trimmed.fastq.gz\n",
      "Approx 40% complete for trunc_SE6151_SA60188_S5_L004_R1_trimmed.fastq.gz\n",
      "Approx 45% complete for trunc_SE6151_SA60188_S5_L004_R1_trimmed.fastq.gz\n",
      "Approx 50% complete for trunc_SE6151_SA60188_S5_L004_R1_trimmed.fastq.gz\n",
      "Approx 55% complete for trunc_SE6151_SA60188_S5_L004_R1_trimmed.fastq.gz\n",
      "Approx 60% complete for trunc_SE6151_SA60188_S5_L004_R1_trimmed.fastq.gz\n",
      "Approx 65% complete for trunc_SE6151_SA60188_S5_L004_R1_trimmed.fastq.gz\n",
      "Approx 70% complete for trunc_SE6151_SA60188_S5_L004_R1_trimmed.fastq.gz\n",
      "Approx 75% complete for trunc_SE6151_SA60188_S5_L004_R1_trimmed.fastq.gz\n",
      "Approx 80% complete for trunc_SE6151_SA60188_S5_L004_R1_trimmed.fastq.gz\n",
      "Approx 85% complete for trunc_SE6151_SA60188_S5_L004_R1_trimmed.fastq.gz\n",
      "Approx 90% complete for trunc_SE6151_SA60188_S5_L004_R1_trimmed.fastq.gz\n",
      "Approx 95% complete for trunc_SE6151_SA60188_S5_L004_R1_trimmed.fastq.gz\n",
      "Analysis complete for trunc_SE6151_SA60188_S5_L004_R1_trimmed.fastq.gz\n",
      "Started analysis of trunc_SE6151_SA60188_S5_L004_R2_trimmed.fastq.gz\n",
      "Approx 5% complete for trunc_SE6151_SA60188_S5_L004_R2_trimmed.fastq.gz\n",
      "Approx 10% complete for trunc_SE6151_SA60188_S5_L004_R2_trimmed.fastq.gz\n",
      "Approx 15% complete for trunc_SE6151_SA60188_S5_L004_R2_trimmed.fastq.gz\n",
      "Approx 20% complete for trunc_SE6151_SA60188_S5_L004_R2_trimmed.fastq.gz\n",
      "Approx 25% complete for trunc_SE6151_SA60188_S5_L004_R2_trimmed.fastq.gz\n",
      "Approx 30% complete for trunc_SE6151_SA60188_S5_L004_R2_trimmed.fastq.gz\n",
      "Approx 35% complete for trunc_SE6151_SA60188_S5_L004_R2_trimmed.fastq.gz\n",
      "Approx 40% complete for trunc_SE6151_SA60188_S5_L004_R2_trimmed.fastq.gz\n",
      "Approx 45% complete for trunc_SE6151_SA60188_S5_L004_R2_trimmed.fastq.gz\n",
      "Approx 50% complete for trunc_SE6151_SA60188_S5_L004_R2_trimmed.fastq.gz\n",
      "Approx 55% complete for trunc_SE6151_SA60188_S5_L004_R2_trimmed.fastq.gz\n",
      "Approx 60% complete for trunc_SE6151_SA60188_S5_L004_R2_trimmed.fastq.gz\n",
      "Approx 65% complete for trunc_SE6151_SA60188_S5_L004_R2_trimmed.fastq.gz\n",
      "Approx 70% complete for trunc_SE6151_SA60188_S5_L004_R2_trimmed.fastq.gz\n",
      "Approx 75% complete for trunc_SE6151_SA60188_S5_L004_R2_trimmed.fastq.gz\n",
      "Approx 80% complete for trunc_SE6151_SA60188_S5_L004_R2_trimmed.fastq.gz\n",
      "Approx 85% complete for trunc_SE6151_SA60188_S5_L004_R2_trimmed.fastq.gz\n",
      "Approx 90% complete for trunc_SE6151_SA60188_S5_L004_R2_trimmed.fastq.gz\n",
      "Approx 95% complete for trunc_SE6151_SA60188_S5_L004_R2_trimmed.fastq.gz\n",
      "Analysis complete for trunc_SE6151_SA60188_S5_L004_R2_trimmed.fastq.gz\n",
      "Started analysis of trunc_SE6151_SA60189_S6_L004_R1_trimmed.fastq.gz\n",
      "Approx 5% complete for trunc_SE6151_SA60189_S6_L004_R1_trimmed.fastq.gz\n",
      "Approx 10% complete for trunc_SE6151_SA60189_S6_L004_R1_trimmed.fastq.gz\n",
      "Approx 15% complete for trunc_SE6151_SA60189_S6_L004_R1_trimmed.fastq.gz\n",
      "Approx 20% complete for trunc_SE6151_SA60189_S6_L004_R1_trimmed.fastq.gz\n",
      "Approx 25% complete for trunc_SE6151_SA60189_S6_L004_R1_trimmed.fastq.gz\n",
      "Approx 30% complete for trunc_SE6151_SA60189_S6_L004_R1_trimmed.fastq.gz\n",
      "Approx 35% complete for trunc_SE6151_SA60189_S6_L004_R1_trimmed.fastq.gz\n",
      "Approx 40% complete for trunc_SE6151_SA60189_S6_L004_R1_trimmed.fastq.gz\n",
      "Approx 45% complete for trunc_SE6151_SA60189_S6_L004_R1_trimmed.fastq.gz\n",
      "Approx 50% complete for trunc_SE6151_SA60189_S6_L004_R1_trimmed.fastq.gz\n",
      "Approx 55% complete for trunc_SE6151_SA60189_S6_L004_R1_trimmed.fastq.gz\n",
      "Approx 60% complete for trunc_SE6151_SA60189_S6_L004_R1_trimmed.fastq.gz\n",
      "Approx 65% complete for trunc_SE6151_SA60189_S6_L004_R1_trimmed.fastq.gz\n",
      "Approx 70% complete for trunc_SE6151_SA60189_S6_L004_R1_trimmed.fastq.gz\n",
      "Approx 75% complete for trunc_SE6151_SA60189_S6_L004_R1_trimmed.fastq.gz\n",
      "Approx 80% complete for trunc_SE6151_SA60189_S6_L004_R1_trimmed.fastq.gz\n",
      "Approx 85% complete for trunc_SE6151_SA60189_S6_L004_R1_trimmed.fastq.gz\n",
      "Approx 90% complete for trunc_SE6151_SA60189_S6_L004_R1_trimmed.fastq.gz\n",
      "Approx 95% complete for trunc_SE6151_SA60189_S6_L004_R1_trimmed.fastq.gz\n",
      "Analysis complete for trunc_SE6151_SA60189_S6_L004_R1_trimmed.fastq.gz\n",
      "Started analysis of trunc_SE6151_SA60189_S6_L004_R2_trimmed.fastq.gz\n",
      "Approx 5% complete for trunc_SE6151_SA60189_S6_L004_R2_trimmed.fastq.gz\n",
      "Approx 10% complete for trunc_SE6151_SA60189_S6_L004_R2_trimmed.fastq.gz\n",
      "Approx 15% complete for trunc_SE6151_SA60189_S6_L004_R2_trimmed.fastq.gz\n",
      "Approx 20% complete for trunc_SE6151_SA60189_S6_L004_R2_trimmed.fastq.gz\n",
      "Approx 25% complete for trunc_SE6151_SA60189_S6_L004_R2_trimmed.fastq.gz\n",
      "Approx 30% complete for trunc_SE6151_SA60189_S6_L004_R2_trimmed.fastq.gz\n",
      "Approx 35% complete for trunc_SE6151_SA60189_S6_L004_R2_trimmed.fastq.gz\n",
      "Approx 40% complete for trunc_SE6151_SA60189_S6_L004_R2_trimmed.fastq.gz\n",
      "Approx 45% complete for trunc_SE6151_SA60189_S6_L004_R2_trimmed.fastq.gz\n",
      "Approx 50% complete for trunc_SE6151_SA60189_S6_L004_R2_trimmed.fastq.gz\n",
      "Approx 55% complete for trunc_SE6151_SA60189_S6_L004_R2_trimmed.fastq.gz\n",
      "Approx 60% complete for trunc_SE6151_SA60189_S6_L004_R2_trimmed.fastq.gz\n",
      "Approx 65% complete for trunc_SE6151_SA60189_S6_L004_R2_trimmed.fastq.gz\n",
      "Approx 70% complete for trunc_SE6151_SA60189_S6_L004_R2_trimmed.fastq.gz\n",
      "Approx 75% complete for trunc_SE6151_SA60189_S6_L004_R2_trimmed.fastq.gz\n",
      "Approx 80% complete for trunc_SE6151_SA60189_S6_L004_R2_trimmed.fastq.gz\n",
      "Approx 85% complete for trunc_SE6151_SA60189_S6_L004_R2_trimmed.fastq.gz\n",
      "Approx 90% complete for trunc_SE6151_SA60189_S6_L004_R2_trimmed.fastq.gz\n",
      "Approx 95% complete for trunc_SE6151_SA60189_S6_L004_R2_trimmed.fastq.gz\n",
      "Analysis complete for trunc_SE6151_SA60189_S6_L004_R2_trimmed.fastq.gz\n",
      "Started analysis of trunc_SE6151_SA60190_S7_L004_R1_trimmed.fastq.gz\n",
      "Approx 35% complete for trunc_SE6151_SA60190_S7_L004_R1_trimmed.fastq.gz\n",
      "Approx 60% complete for trunc_SE6151_SA60190_S7_L004_R1_trimmed.fastq.gz\n",
      "Approx 85% complete for trunc_SE6151_SA60190_S7_L004_R1_trimmed.fastq.gz\n",
      "Analysis complete for trunc_SE6151_SA60190_S7_L004_R1_trimmed.fastq.gz\n",
      "Started analysis of trunc_SE6151_SA60190_S7_L004_R2_trimmed.fastq.gz\n",
      "Approx 30% complete for trunc_SE6151_SA60190_S7_L004_R2_trimmed.fastq.gz\n",
      "Approx 60% complete for trunc_SE6151_SA60190_S7_L004_R2_trimmed.fastq.gz\n",
      "Approx 90% complete for trunc_SE6151_SA60190_S7_L004_R2_trimmed.fastq.gz\n",
      "Analysis complete for trunc_SE6151_SA60190_S7_L004_R2_trimmed.fastq.gz\n",
      "Started analysis of trunc_SE6151_SA60191_S8_L004_R1_trimmed.fastq.gz\n",
      "Approx 50% complete for trunc_SE6151_SA60191_S8_L004_R1_trimmed.fastq.gz\n",
      "Approx 85% complete for trunc_SE6151_SA60191_S8_L004_R1_trimmed.fastq.gz\n",
      "Analysis complete for trunc_SE6151_SA60191_S8_L004_R1_trimmed.fastq.gz\n",
      "Started analysis of trunc_SE6151_SA60191_S8_L004_R2_trimmed.fastq.gz\n",
      "Approx 45% complete for trunc_SE6151_SA60191_S8_L004_R2_trimmed.fastq.gz\n",
      "Approx 85% complete for trunc_SE6151_SA60191_S8_L004_R2_trimmed.fastq.gz\n",
      "Analysis complete for trunc_SE6151_SA60191_S8_L004_R2_trimmed.fastq.gz\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "\n",
       "        <iframe\n",
       "            width=\"800\"\n",
       "            height=\"600\"\n",
       "            src=\"./data/fastqc/trunc_SE6151_SA60184_S1_L004_R1_trimmed_fastqc.html\"\n",
       "            frameborder=\"0\"\n",
       "            allowfullscreen\n",
       "            \n",
       "        ></iframe>\n",
       "        "
      ],
      "text/plain": [
       "<IPython.lib.display.IFrame at 0x7fcbbdde6350>"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Run fastqc\n",
    "!fastqc -o data/fastqc data/trimmed/*_trimmed.fastq.gz\n",
    "\n",
    "# Example of in-window display of fastqc output file (SA60184).\n",
    "from IPython.display import IFrame\n",
    "IFrame(src='./data/fastqc/trunc_SE6151_SA60184_S1_L004_R1_trimmed_fastqc.html', width=800, height=600)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "d8068f13-8cf5-4381-8485-81f061456d67",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "  \u001b[34m/\u001b[0m\u001b[32m/\u001b[0m\u001b[31m/\u001b[0m \u001b]8;id=336355;https://multiqc.info\u001b\\\u001b[1mMultiQC\u001b[0m\u001b]8;;\u001b\\ 🔍 \u001b[2m| v1.13\u001b[0m\n",
      "\n",
      "\u001b[34m|           multiqc\u001b[0m | Search path : /home/jupyter/Example_Whim_Notebook/data/fastqc\n",
      "\u001b[2K\u001b[34m|\u001b[0m         \u001b[34msearching\u001b[0m | \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[35m100%\u001b[0m \u001b[32m32/32\u001b[0m  \u001b[0m0m  \n",
      "\u001b[?25h\u001b[34m|            fastqc\u001b[0m | Found 16 reports\n",
      "\u001b[34m|           multiqc\u001b[0m | Compressing plot data\n",
      "\u001b[34m|           multiqc\u001b[0m | \u001b[33mDeleting    : data/multiqc/multiqc_report.html   (-f was specified)\u001b[0m\n",
      "\u001b[34m|           multiqc\u001b[0m | \u001b[33mDeleting    : data/multiqc/multiqc_data   (-f was specified)\u001b[0m\n",
      "\u001b[34m|           multiqc\u001b[0m | Report      : data/multiqc/multiqc_report.html\n",
      "\u001b[34m|           multiqc\u001b[0m | Data        : data/multiqc/multiqc_data\n",
      "\u001b[34m|           multiqc\u001b[0m | MultiQC complete\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Sample</th>\n",
       "      <th>Filename</th>\n",
       "      <th>File type</th>\n",
       "      <th>Encoding</th>\n",
       "      <th>Total Sequences</th>\n",
       "      <th>Sequences flagged as poor quality</th>\n",
       "      <th>Sequence length</th>\n",
       "      <th>%GC</th>\n",
       "      <th>total_deduplicated_percentage</th>\n",
       "      <th>avg_sequence_length</th>\n",
       "      <th>...</th>\n",
       "      <th>per_base_sequence_quality</th>\n",
       "      <th>per_tile_sequence_quality</th>\n",
       "      <th>per_sequence_quality_scores</th>\n",
       "      <th>per_base_sequence_content</th>\n",
       "      <th>per_sequence_gc_content</th>\n",
       "      <th>per_base_n_content</th>\n",
       "      <th>sequence_length_distribution</th>\n",
       "      <th>sequence_duplication_levels</th>\n",
       "      <th>overrepresented_sequences</th>\n",
       "      <th>adapter_content</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>trunc_SE6151_SA60184_S1_L004_R1</td>\n",
       "      <td>trunc_SE6151_SA60184_S1_L004_R1_trimmed.fastq.gz</td>\n",
       "      <td>Conventional base calls</td>\n",
       "      <td>Sanger / Illumina 1.9</td>\n",
       "      <td>21329.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>150-151</td>\n",
       "      <td>38.0</td>\n",
       "      <td>78.194008</td>\n",
       "      <td>150.992545</td>\n",
       "      <td>...</td>\n",
       "      <td>pass</td>\n",
       "      <td>warn</td>\n",
       "      <td>pass</td>\n",
       "      <td>fail</td>\n",
       "      <td>warn</td>\n",
       "      <td>pass</td>\n",
       "      <td>warn</td>\n",
       "      <td>pass</td>\n",
       "      <td>warn</td>\n",
       "      <td>pass</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>trunc_SE6151_SA60184_S1_L004_R2</td>\n",
       "      <td>trunc_SE6151_SA60184_S1_L004_R2_trimmed.fastq.gz</td>\n",
       "      <td>Conventional base calls</td>\n",
       "      <td>Sanger / Illumina 1.9</td>\n",
       "      <td>21329.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>150-151</td>\n",
       "      <td>41.0</td>\n",
       "      <td>87.866285</td>\n",
       "      <td>150.987201</td>\n",
       "      <td>...</td>\n",
       "      <td>pass</td>\n",
       "      <td>warn</td>\n",
       "      <td>pass</td>\n",
       "      <td>warn</td>\n",
       "      <td>pass</td>\n",
       "      <td>pass</td>\n",
       "      <td>warn</td>\n",
       "      <td>pass</td>\n",
       "      <td>warn</td>\n",
       "      <td>pass</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>trunc_SE6151_SA60185_S2_L004_R1</td>\n",
       "      <td>trunc_SE6151_SA60185_S2_L004_R1_trimmed.fastq.gz</td>\n",
       "      <td>Conventional base calls</td>\n",
       "      <td>Sanger / Illumina 1.9</td>\n",
       "      <td>19635.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>150-151</td>\n",
       "      <td>37.0</td>\n",
       "      <td>85.663356</td>\n",
       "      <td>150.991851</td>\n",
       "      <td>...</td>\n",
       "      <td>pass</td>\n",
       "      <td>warn</td>\n",
       "      <td>pass</td>\n",
       "      <td>fail</td>\n",
       "      <td>warn</td>\n",
       "      <td>pass</td>\n",
       "      <td>warn</td>\n",
       "      <td>pass</td>\n",
       "      <td>warn</td>\n",
       "      <td>pass</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>trunc_SE6151_SA60185_S2_L004_R2</td>\n",
       "      <td>trunc_SE6151_SA60185_S2_L004_R2_trimmed.fastq.gz</td>\n",
       "      <td>Conventional base calls</td>\n",
       "      <td>Sanger / Illumina 1.9</td>\n",
       "      <td>19635.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>150-151</td>\n",
       "      <td>39.0</td>\n",
       "      <td>93.027757</td>\n",
       "      <td>150.986096</td>\n",
       "      <td>...</td>\n",
       "      <td>pass</td>\n",
       "      <td>warn</td>\n",
       "      <td>pass</td>\n",
       "      <td>warn</td>\n",
       "      <td>warn</td>\n",
       "      <td>pass</td>\n",
       "      <td>warn</td>\n",
       "      <td>pass</td>\n",
       "      <td>pass</td>\n",
       "      <td>pass</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>trunc_SE6151_SA60186_S3_L004_R1</td>\n",
       "      <td>trunc_SE6151_SA60186_S3_L004_R1_trimmed.fastq.gz</td>\n",
       "      <td>Conventional base calls</td>\n",
       "      <td>Sanger / Illumina 1.9</td>\n",
       "      <td>17563.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>150-151</td>\n",
       "      <td>40.0</td>\n",
       "      <td>81.500883</td>\n",
       "      <td>150.993224</td>\n",
       "      <td>...</td>\n",
       "      <td>pass</td>\n",
       "      <td>warn</td>\n",
       "      <td>pass</td>\n",
       "      <td>fail</td>\n",
       "      <td>fail</td>\n",
       "      <td>pass</td>\n",
       "      <td>warn</td>\n",
       "      <td>pass</td>\n",
       "      <td>warn</td>\n",
       "      <td>pass</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>trunc_SE6151_SA60186_S3_L004_R2</td>\n",
       "      <td>trunc_SE6151_SA60186_S3_L004_R2_trimmed.fastq.gz</td>\n",
       "      <td>Conventional base calls</td>\n",
       "      <td>Sanger / Illumina 1.9</td>\n",
       "      <td>17563.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>150-151</td>\n",
       "      <td>42.0</td>\n",
       "      <td>90.229460</td>\n",
       "      <td>150.987474</td>\n",
       "      <td>...</td>\n",
       "      <td>pass</td>\n",
       "      <td>fail</td>\n",
       "      <td>pass</td>\n",
       "      <td>warn</td>\n",
       "      <td>warn</td>\n",
       "      <td>pass</td>\n",
       "      <td>warn</td>\n",
       "      <td>pass</td>\n",
       "      <td>warn</td>\n",
       "      <td>pass</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>trunc_SE6151_SA60187_S4_L004_R1</td>\n",
       "      <td>trunc_SE6151_SA60187_S4_L004_R1_trimmed.fastq.gz</td>\n",
       "      <td>Conventional base calls</td>\n",
       "      <td>Sanger / Illumina 1.9</td>\n",
       "      <td>21689.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>150-151</td>\n",
       "      <td>35.0</td>\n",
       "      <td>77.762921</td>\n",
       "      <td>150.991885</td>\n",
       "      <td>...</td>\n",
       "      <td>pass</td>\n",
       "      <td>warn</td>\n",
       "      <td>pass</td>\n",
       "      <td>fail</td>\n",
       "      <td>warn</td>\n",
       "      <td>pass</td>\n",
       "      <td>warn</td>\n",
       "      <td>pass</td>\n",
       "      <td>warn</td>\n",
       "      <td>pass</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>trunc_SE6151_SA60187_S4_L004_R2</td>\n",
       "      <td>trunc_SE6151_SA60187_S4_L004_R2_trimmed.fastq.gz</td>\n",
       "      <td>Conventional base calls</td>\n",
       "      <td>Sanger / Illumina 1.9</td>\n",
       "      <td>21689.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>150-151</td>\n",
       "      <td>39.0</td>\n",
       "      <td>88.288994</td>\n",
       "      <td>150.986030</td>\n",
       "      <td>...</td>\n",
       "      <td>pass</td>\n",
       "      <td>warn</td>\n",
       "      <td>pass</td>\n",
       "      <td>warn</td>\n",
       "      <td>warn</td>\n",
       "      <td>pass</td>\n",
       "      <td>warn</td>\n",
       "      <td>pass</td>\n",
       "      <td>warn</td>\n",
       "      <td>pass</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>trunc_SE6151_SA60188_S5_L004_R1</td>\n",
       "      <td>trunc_SE6151_SA60188_S5_L004_R1_trimmed.fastq.gz</td>\n",
       "      <td>Conventional base calls</td>\n",
       "      <td>Sanger / Illumina 1.9</td>\n",
       "      <td>24736.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>150-151</td>\n",
       "      <td>80.0</td>\n",
       "      <td>3.553525</td>\n",
       "      <td>150.990944</td>\n",
       "      <td>...</td>\n",
       "      <td>pass</td>\n",
       "      <td>warn</td>\n",
       "      <td>pass</td>\n",
       "      <td>fail</td>\n",
       "      <td>fail</td>\n",
       "      <td>pass</td>\n",
       "      <td>warn</td>\n",
       "      <td>fail</td>\n",
       "      <td>fail</td>\n",
       "      <td>pass</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>trunc_SE6151_SA60188_S5_L004_R2</td>\n",
       "      <td>trunc_SE6151_SA60188_S5_L004_R2_trimmed.fastq.gz</td>\n",
       "      <td>Conventional base calls</td>\n",
       "      <td>Sanger / Illumina 1.9</td>\n",
       "      <td>24736.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>150-151</td>\n",
       "      <td>76.0</td>\n",
       "      <td>20.015362</td>\n",
       "      <td>150.986376</td>\n",
       "      <td>...</td>\n",
       "      <td>pass</td>\n",
       "      <td>fail</td>\n",
       "      <td>pass</td>\n",
       "      <td>fail</td>\n",
       "      <td>fail</td>\n",
       "      <td>pass</td>\n",
       "      <td>warn</td>\n",
       "      <td>fail</td>\n",
       "      <td>fail</td>\n",
       "      <td>pass</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>trunc_SE6151_SA60189_S6_L004_R1</td>\n",
       "      <td>trunc_SE6151_SA60189_S6_L004_R1_trimmed.fastq.gz</td>\n",
       "      <td>Conventional base calls</td>\n",
       "      <td>Sanger / Illumina 1.9</td>\n",
       "      <td>20582.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>150-151</td>\n",
       "      <td>39.0</td>\n",
       "      <td>82.057137</td>\n",
       "      <td>150.991449</td>\n",
       "      <td>...</td>\n",
       "      <td>pass</td>\n",
       "      <td>warn</td>\n",
       "      <td>pass</td>\n",
       "      <td>fail</td>\n",
       "      <td>pass</td>\n",
       "      <td>pass</td>\n",
       "      <td>warn</td>\n",
       "      <td>pass</td>\n",
       "      <td>warn</td>\n",
       "      <td>pass</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>trunc_SE6151_SA60189_S6_L004_R2</td>\n",
       "      <td>trunc_SE6151_SA60189_S6_L004_R2_trimmed.fastq.gz</td>\n",
       "      <td>Conventional base calls</td>\n",
       "      <td>Sanger / Illumina 1.9</td>\n",
       "      <td>20582.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>150-151</td>\n",
       "      <td>42.0</td>\n",
       "      <td>90.384802</td>\n",
       "      <td>150.986687</td>\n",
       "      <td>...</td>\n",
       "      <td>pass</td>\n",
       "      <td>warn</td>\n",
       "      <td>pass</td>\n",
       "      <td>warn</td>\n",
       "      <td>pass</td>\n",
       "      <td>pass</td>\n",
       "      <td>warn</td>\n",
       "      <td>pass</td>\n",
       "      <td>warn</td>\n",
       "      <td>pass</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>trunc_SE6151_SA60190_S7_L004_R1</td>\n",
       "      <td>trunc_SE6151_SA60190_S7_L004_R1_trimmed.fastq.gz</td>\n",
       "      <td>Conventional base calls</td>\n",
       "      <td>Sanger / Illumina 1.9</td>\n",
       "      <td>3281.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>150-151</td>\n",
       "      <td>40.0</td>\n",
       "      <td>26.668699</td>\n",
       "      <td>150.994209</td>\n",
       "      <td>...</td>\n",
       "      <td>pass</td>\n",
       "      <td>fail</td>\n",
       "      <td>pass</td>\n",
       "      <td>fail</td>\n",
       "      <td>fail</td>\n",
       "      <td>pass</td>\n",
       "      <td>warn</td>\n",
       "      <td>fail</td>\n",
       "      <td>fail</td>\n",
       "      <td>warn</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>trunc_SE6151_SA60190_S7_L004_R2</td>\n",
       "      <td>trunc_SE6151_SA60190_S7_L004_R2_trimmed.fastq.gz</td>\n",
       "      <td>Conventional base calls</td>\n",
       "      <td>Sanger / Illumina 1.9</td>\n",
       "      <td>3281.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>150-151</td>\n",
       "      <td>44.0</td>\n",
       "      <td>51.752514</td>\n",
       "      <td>150.990247</td>\n",
       "      <td>...</td>\n",
       "      <td>pass</td>\n",
       "      <td>fail</td>\n",
       "      <td>pass</td>\n",
       "      <td>warn</td>\n",
       "      <td>fail</td>\n",
       "      <td>pass</td>\n",
       "      <td>warn</td>\n",
       "      <td>warn</td>\n",
       "      <td>fail</td>\n",
       "      <td>warn</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>trunc_SE6151_SA60191_S8_L004_R1</td>\n",
       "      <td>trunc_SE6151_SA60191_S8_L004_R1_trimmed.fastq.gz</td>\n",
       "      <td>Conventional base calls</td>\n",
       "      <td>Sanger / Illumina 1.9</td>\n",
       "      <td>2328.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>150-151</td>\n",
       "      <td>51.0</td>\n",
       "      <td>48.152921</td>\n",
       "      <td>150.990979</td>\n",
       "      <td>...</td>\n",
       "      <td>pass</td>\n",
       "      <td>fail</td>\n",
       "      <td>pass</td>\n",
       "      <td>fail</td>\n",
       "      <td>fail</td>\n",
       "      <td>pass</td>\n",
       "      <td>warn</td>\n",
       "      <td>fail</td>\n",
       "      <td>fail</td>\n",
       "      <td>fail</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>trunc_SE6151_SA60191_S8_L004_R2</td>\n",
       "      <td>trunc_SE6151_SA60191_S8_L004_R2_trimmed.fastq.gz</td>\n",
       "      <td>Conventional base calls</td>\n",
       "      <td>Sanger / Illumina 1.9</td>\n",
       "      <td>2328.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>150-151</td>\n",
       "      <td>55.0</td>\n",
       "      <td>44.029210</td>\n",
       "      <td>150.986254</td>\n",
       "      <td>...</td>\n",
       "      <td>pass</td>\n",
       "      <td>fail</td>\n",
       "      <td>pass</td>\n",
       "      <td>fail</td>\n",
       "      <td>fail</td>\n",
       "      <td>pass</td>\n",
       "      <td>warn</td>\n",
       "      <td>fail</td>\n",
       "      <td>fail</td>\n",
       "      <td>fail</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>16 rows × 21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                             Sample  \\\n",
       "0   trunc_SE6151_SA60184_S1_L004_R1   \n",
       "1   trunc_SE6151_SA60184_S1_L004_R2   \n",
       "2   trunc_SE6151_SA60185_S2_L004_R1   \n",
       "3   trunc_SE6151_SA60185_S2_L004_R2   \n",
       "4   trunc_SE6151_SA60186_S3_L004_R1   \n",
       "5   trunc_SE6151_SA60186_S3_L004_R2   \n",
       "6   trunc_SE6151_SA60187_S4_L004_R1   \n",
       "7   trunc_SE6151_SA60187_S4_L004_R2   \n",
       "8   trunc_SE6151_SA60188_S5_L004_R1   \n",
       "9   trunc_SE6151_SA60188_S5_L004_R2   \n",
       "10  trunc_SE6151_SA60189_S6_L004_R1   \n",
       "11  trunc_SE6151_SA60189_S6_L004_R2   \n",
       "12  trunc_SE6151_SA60190_S7_L004_R1   \n",
       "13  trunc_SE6151_SA60190_S7_L004_R2   \n",
       "14  trunc_SE6151_SA60191_S8_L004_R1   \n",
       "15  trunc_SE6151_SA60191_S8_L004_R2   \n",
       "\n",
       "                                            Filename                File type  \\\n",
       "0   trunc_SE6151_SA60184_S1_L004_R1_trimmed.fastq.gz  Conventional base calls   \n",
       "1   trunc_SE6151_SA60184_S1_L004_R2_trimmed.fastq.gz  Conventional base calls   \n",
       "2   trunc_SE6151_SA60185_S2_L004_R1_trimmed.fastq.gz  Conventional base calls   \n",
       "3   trunc_SE6151_SA60185_S2_L004_R2_trimmed.fastq.gz  Conventional base calls   \n",
       "4   trunc_SE6151_SA60186_S3_L004_R1_trimmed.fastq.gz  Conventional base calls   \n",
       "5   trunc_SE6151_SA60186_S3_L004_R2_trimmed.fastq.gz  Conventional base calls   \n",
       "6   trunc_SE6151_SA60187_S4_L004_R1_trimmed.fastq.gz  Conventional base calls   \n",
       "7   trunc_SE6151_SA60187_S4_L004_R2_trimmed.fastq.gz  Conventional base calls   \n",
       "8   trunc_SE6151_SA60188_S5_L004_R1_trimmed.fastq.gz  Conventional base calls   \n",
       "9   trunc_SE6151_SA60188_S5_L004_R2_trimmed.fastq.gz  Conventional base calls   \n",
       "10  trunc_SE6151_SA60189_S6_L004_R1_trimmed.fastq.gz  Conventional base calls   \n",
       "11  trunc_SE6151_SA60189_S6_L004_R2_trimmed.fastq.gz  Conventional base calls   \n",
       "12  trunc_SE6151_SA60190_S7_L004_R1_trimmed.fastq.gz  Conventional base calls   \n",
       "13  trunc_SE6151_SA60190_S7_L004_R2_trimmed.fastq.gz  Conventional base calls   \n",
       "14  trunc_SE6151_SA60191_S8_L004_R1_trimmed.fastq.gz  Conventional base calls   \n",
       "15  trunc_SE6151_SA60191_S8_L004_R2_trimmed.fastq.gz  Conventional base calls   \n",
       "\n",
       "                 Encoding  Total Sequences  Sequences flagged as poor quality  \\\n",
       "0   Sanger / Illumina 1.9          21329.0                                0.0   \n",
       "1   Sanger / Illumina 1.9          21329.0                                0.0   \n",
       "2   Sanger / Illumina 1.9          19635.0                                0.0   \n",
       "3   Sanger / Illumina 1.9          19635.0                                0.0   \n",
       "4   Sanger / Illumina 1.9          17563.0                                0.0   \n",
       "5   Sanger / Illumina 1.9          17563.0                                0.0   \n",
       "6   Sanger / Illumina 1.9          21689.0                                0.0   \n",
       "7   Sanger / Illumina 1.9          21689.0                                0.0   \n",
       "8   Sanger / Illumina 1.9          24736.0                                0.0   \n",
       "9   Sanger / Illumina 1.9          24736.0                                0.0   \n",
       "10  Sanger / Illumina 1.9          20582.0                                0.0   \n",
       "11  Sanger / Illumina 1.9          20582.0                                0.0   \n",
       "12  Sanger / Illumina 1.9           3281.0                                0.0   \n",
       "13  Sanger / Illumina 1.9           3281.0                                0.0   \n",
       "14  Sanger / Illumina 1.9           2328.0                                0.0   \n",
       "15  Sanger / Illumina 1.9           2328.0                                0.0   \n",
       "\n",
       "   Sequence length   %GC  total_deduplicated_percentage  avg_sequence_length  \\\n",
       "0          150-151  38.0                      78.194008           150.992545   \n",
       "1          150-151  41.0                      87.866285           150.987201   \n",
       "2          150-151  37.0                      85.663356           150.991851   \n",
       "3          150-151  39.0                      93.027757           150.986096   \n",
       "4          150-151  40.0                      81.500883           150.993224   \n",
       "5          150-151  42.0                      90.229460           150.987474   \n",
       "6          150-151  35.0                      77.762921           150.991885   \n",
       "7          150-151  39.0                      88.288994           150.986030   \n",
       "8          150-151  80.0                       3.553525           150.990944   \n",
       "9          150-151  76.0                      20.015362           150.986376   \n",
       "10         150-151  39.0                      82.057137           150.991449   \n",
       "11         150-151  42.0                      90.384802           150.986687   \n",
       "12         150-151  40.0                      26.668699           150.994209   \n",
       "13         150-151  44.0                      51.752514           150.990247   \n",
       "14         150-151  51.0                      48.152921           150.990979   \n",
       "15         150-151  55.0                      44.029210           150.986254   \n",
       "\n",
       "    ... per_base_sequence_quality per_tile_sequence_quality  \\\n",
       "0   ...                      pass                      warn   \n",
       "1   ...                      pass                      warn   \n",
       "2   ...                      pass                      warn   \n",
       "3   ...                      pass                      warn   \n",
       "4   ...                      pass                      warn   \n",
       "5   ...                      pass                      fail   \n",
       "6   ...                      pass                      warn   \n",
       "7   ...                      pass                      warn   \n",
       "8   ...                      pass                      warn   \n",
       "9   ...                      pass                      fail   \n",
       "10  ...                      pass                      warn   \n",
       "11  ...                      pass                      warn   \n",
       "12  ...                      pass                      fail   \n",
       "13  ...                      pass                      fail   \n",
       "14  ...                      pass                      fail   \n",
       "15  ...                      pass                      fail   \n",
       "\n",
       "   per_sequence_quality_scores per_base_sequence_content  \\\n",
       "0                         pass                      fail   \n",
       "1                         pass                      warn   \n",
       "2                         pass                      fail   \n",
       "3                         pass                      warn   \n",
       "4                         pass                      fail   \n",
       "5                         pass                      warn   \n",
       "6                         pass                      fail   \n",
       "7                         pass                      warn   \n",
       "8                         pass                      fail   \n",
       "9                         pass                      fail   \n",
       "10                        pass                      fail   \n",
       "11                        pass                      warn   \n",
       "12                        pass                      fail   \n",
       "13                        pass                      warn   \n",
       "14                        pass                      fail   \n",
       "15                        pass                      fail   \n",
       "\n",
       "   per_sequence_gc_content per_base_n_content sequence_length_distribution  \\\n",
       "0                     warn               pass                         warn   \n",
       "1                     pass               pass                         warn   \n",
       "2                     warn               pass                         warn   \n",
       "3                     warn               pass                         warn   \n",
       "4                     fail               pass                         warn   \n",
       "5                     warn               pass                         warn   \n",
       "6                     warn               pass                         warn   \n",
       "7                     warn               pass                         warn   \n",
       "8                     fail               pass                         warn   \n",
       "9                     fail               pass                         warn   \n",
       "10                    pass               pass                         warn   \n",
       "11                    pass               pass                         warn   \n",
       "12                    fail               pass                         warn   \n",
       "13                    fail               pass                         warn   \n",
       "14                    fail               pass                         warn   \n",
       "15                    fail               pass                         warn   \n",
       "\n",
       "   sequence_duplication_levels overrepresented_sequences adapter_content  \n",
       "0                         pass                      warn            pass  \n",
       "1                         pass                      warn            pass  \n",
       "2                         pass                      warn            pass  \n",
       "3                         pass                      pass            pass  \n",
       "4                         pass                      warn            pass  \n",
       "5                         pass                      warn            pass  \n",
       "6                         pass                      warn            pass  \n",
       "7                         pass                      warn            pass  \n",
       "8                         fail                      fail            pass  \n",
       "9                         fail                      fail            pass  \n",
       "10                        pass                      warn            pass  \n",
       "11                        pass                      warn            pass  \n",
       "12                        fail                      fail            warn  \n",
       "13                        warn                      fail            warn  \n",
       "14                        fail                      fail            fail  \n",
       "15                        fail                      fail            fail  \n",
       "\n",
       "[16 rows x 21 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "!multiqc -f data/fastqc -o data/multiqc\n",
    "\n",
    "import pandas as pd\n",
    "dframe = pd.read_csv(\"data/multiqc/multiqc_data/multiqc_fastqc.txt\", sep='\\t')\n",
    "display(dframe)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "54262a2d-ec30-4ad9-8f49-63db2c41e54d",
   "metadata": {},
   "source": [
    "# Rsem\n",
    "\n",
    "In order to map reads to genes, and quantify read counts, an index must be created. In this case, we are using rsem for index creation and quantifying read counts.\n",
    "\n",
    "The output and error log for this software is too large to display in the cell below, so is stored in a text log file 'rsem/rsem_log.txt' instead. \n",
    "\n",
    "For the example truncated set this can finish within a few minutes, but with the actual data set this will take around eight hours to fully run."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "0953c9f6-083d-4b71-add6-fbdb27c6d91f",
   "metadata": {
    "scrolled": true,
    "tags": []
   },
   "outputs": [],
   "source": [
    "## Build transcriptome reference\n",
    "\n",
    "#!rsem-prepare-reference --gtf data/reference/GRCz11.108.gtf --bowtie -p 8 data/reference/GRCz11.dna_sm.toplevel.fa data/reference/zebrafish_rsem/GRCz11_108 "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "acd76c1a-2b41-4bc6-b6d5-63cda85785ab",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "running calculate counts...\n",
      "finished\n"
     ]
    }
   ],
   "source": [
    "# to make writing commands easier,\n",
    "# again we will make a list and 'pipe' the list\n",
    "# into rsem.\n",
    "# first make a list\n",
    "!ls data/trimmed/*R1*trimmed* |  sed 's|data/trimmed/||g' | sed 's|.fastq.gz||g' > trimmed_ids.txt\n",
    "# then run rsem\n",
    "!echo 'running calculate counts...'\n",
    "!cat trimmed_ids.txt |  xargs -I {} rsem-calculate-expression -p 16 --forward-prob 0 data/trimmed/{}.fastq.gz data/reference/zebrafish_rsem/GRCz11_108 data/rsem/{} >data/rsem/rsem_log.txt 2>&1\n",
    "!echo 'finished'"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d0ad4d94-ce11-4116-ac76-2a17309516d3",
   "metadata": {},
   "source": [
    "# Read count file outputs\n",
    "\n",
    "Rsem outputs readcount files (XXXXX.genes.results) which are tables that contain various information, including readcounts in transcripts per kilobase million (TPM) and fragments per kilobase of exon per million mapped fragments (FPKM) format."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "5e5b7af5-9a20-4bf7-a42b-a75c0a9918f8",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "gene_id  transcript_id(s)  length  effective_length  expected_count  TPM  FPKM\n",
      "ENSDARG00000114494  ENSDART00000187218  165.00   46.58    102081.87  128401.41  170718.92\n",
      "ENSDARG00000087732  ENSDART00000122203  298.00   179.58   358926.10  117105.31  155699.95\n",
      "ENSDARG00000110859  ENSDART00000193467  165.00   46.58    26641.94   33510.97   44555.24\n",
      "ENSDARG00000109433  ENSDART00000180315  165.00   46.58    26641.94   33510.97   44555.24\n",
      "ENSDARG00000083063  ENSDART00000118247  165.00   46.58    26641.94   33510.97   44555.24\n",
      "ENSDARG00000081270  ENSDART00000117461  322.00   203.58   87739.30   25251.60   33573.82\n",
      "ENSDARG00000089382  ENSDART00000126982  1212.00  1093.58  467080.56  25024.96   33272.48\n",
      "ENSDARG00000082008  ENSDART00000116893  96.00    6.95     2714.12    22876.81   30416.37\n",
      "ENSDARG00000081580  ENSDART00000116507  72.00    1.95     723.00     21707.91   28862.23\n",
      "ENSDARG00000082753  ENSDART00000115891  952.00   833.58   285374.88  20058.59   26669.34\n"
     ]
    }
   ],
   "source": [
    "# Example of read count table output\n",
    "# (sorted by top 10 expressed genes)\n",
    "!head -1 $(ls data/rsem/*.genes.results | head -n1) | column -t\n",
    "!sort -nrk 6 $(ls data/rsem/*.genes.results | head -n1) |  column -t | head -10  | column -t"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c035c038-5fdd-4b3b-ac3c-023c3db4601c",
   "metadata": {},
   "source": [
    "# Upload files to bucket\n",
    "In order for the files to be shared and downloaded, we need to upload them to the google bucket."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "420d72a2-4e13-434d-8f48-5e3f64fd618a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Copying file://data/rsem/SE8159_SA161910_S1_L007_R1_trimmed.genes.results [Content-Type=application/octet-stream]...\n",
      "Copying file://data/rsem/SE8159_SA161911_S2_L007_R1_trimmed.genes.results [Content-Type=application/octet-stream]...\n",
      "Copying file://data/rsem/SE8159_SA161912_S3_L007_R1_trimmed.genes.results [Content-Type=application/octet-stream]...\n",
      "Copying file://data/rsem/SE8159_SA161913_S4_L007_R1_trimmed.genes.results [Content-Type=application/octet-stream]...\n",
      "- [4 files][ 10.6 MiB/ 10.6 MiB]                                                \n",
      "==> NOTE: You are performing a sequence of gsutil operations that may\n",
      "run significantly faster if you instead use gsutil -m cp ... Please\n",
      "see the -m section under \"gsutil help options\" for further information\n",
      "about when gsutil -m can be advantageous.\n",
      "\n",
      "Copying file://data/rsem/SE8159_SA161914_S5_L007_R1_trimmed.genes.results [Content-Type=application/octet-stream]...\n",
      "Copying file://data/rsem/SE8159_SA161915_S6_L007_R1_trimmed.genes.results [Content-Type=application/octet-stream]...\n",
      "Copying file://data/rsem/SE8159_SA161916_S7_L007_R1_trimmed.genes.results [Content-Type=application/octet-stream]...\n",
      "Copying file://data/rsem/SE8159_SA161917_S8_L007_R1_trimmed.genes.results [Content-Type=application/octet-stream]...\n",
      "Copying file://data/rsem/SE8159_SA161918_S9_L007_R1_trimmed.genes.results [Content-Type=application/octet-stream]...\n",
      "Copying file://data/rsem/SE8159_SA161919_S10_L007_R1_trimmed.genes.results [Content-Type=application/octet-stream]...\n",
      "Copying file://data/rsem/SE8159_SA161920_S11_L007_R1_trimmed.genes.results [Content-Type=application/octet-stream]...\n",
      "Copying file://data/rsem/SE8159_SA161921_S12_L007_R1_trimmed.genes.results [Content-Type=application/octet-stream]...\n",
      "Copying file://data/rsem/SE8159_SA161922_S13_L007_R1_trimmed.genes.results [Content-Type=application/octet-stream]...\n",
      "Copying file://data/rsem/SE8159_SA161923_S14_L007_R1_trimmed.genes.results [Content-Type=application/octet-stream]...\n",
      "Copying file://data/rsem/SE8159_SA161924_S15_L007_R1_trimmed.genes.results [Content-Type=application/octet-stream]...\n",
      "Copying file://data/rsem/SE8161_SA161878_S1_L007_R1_trimmed.genes.results [Content-Type=application/octet-stream]...\n",
      "Copying file://data/rsem/SE8161_SA161879_S2_L007_R1_trimmed.genes.results [Content-Type=application/octet-stream]...\n",
      "Copying file://data/rsem/SE8161_SA161880_S3_L007_R1_trimmed.genes.results [Content-Type=application/octet-stream]...\n",
      "Copying file://data/rsem/SE8161_SA161881_S4_L007_R1_trimmed.genes.results [Content-Type=application/octet-stream]...\n",
      "Copying file://data/rsem/SE8161_SA161882_S5_L007_R1_trimmed.genes.results [Content-Type=application/octet-stream]...\n",
      "Copying file://data/rsem/SE8161_SA161883_S6_L007_R1_trimmed.genes.results [Content-Type=application/octet-stream]...\n",
      "Copying file://data/rsem/SE8161_SA161884_S7_L007_R1_trimmed.genes.results [Content-Type=application/octet-stream]...\n",
      "Copying file://data/rsem/SE8161_SA161885_S8_L007_R1_trimmed.genes.results [Content-Type=application/octet-stream]...\n",
      "Copying file://data/rsem/SE8161_SA161886_S9_L007_R1_trimmed.genes.results [Content-Type=application/octet-stream]...\n",
      "Copying file://data/rsem/trunc_SE6151_SA60184_S1_L004_R1_trimmed.genes.results [Content-Type=application/octet-stream]...\n",
      "Copying file://data/rsem/trunc_SE6151_SA60185_S2_L004_R1_trimmed.genes.results [Content-Type=application/octet-stream]...\n",
      "Copying file://data/rsem/trunc_SE6151_SA60186_S3_L004_R1_trimmed.genes.results [Content-Type=application/octet-stream]...\n",
      "Copying file://data/rsem/trunc_SE6151_SA60187_S4_L004_R1_trimmed.genes.results [Content-Type=application/octet-stream]...\n",
      "Copying file://data/rsem/trunc_SE6151_SA60188_S5_L004_R1_trimmed.genes.results [Content-Type=application/octet-stream]...\n",
      "Copying file://data/rsem/trunc_SE6151_SA60189_S6_L004_R1_trimmed.genes.results [Content-Type=application/octet-stream]...\n",
      "Copying file://data/rsem/trunc_SE6151_SA60190_S7_L004_R1_trimmed.genes.results [Content-Type=application/octet-stream]...\n",
      "Copying file://data/rsem/trunc_SE6151_SA60191_S8_L004_R1_trimmed.genes.results [Content-Type=application/octet-stream]...\n",
      "| [32 files][ 84.7 MiB/ 84.7 MiB]    8.0 MiB/s                                  \n",
      "==> NOTE: You are performing a sequence of gsutil operations that may\n",
      "run significantly faster if you instead use gsutil -m cp ... Please\n",
      "see the -m section under \"gsutil help options\" for further information\n",
      "about when gsutil -m can be advantageous.\n",
      "\n",
      "\n",
      "Operation completed over 32 objects/84.7 MiB.                                    \n"
     ]
    }
   ],
   "source": [
    "## Upload to bucket\n",
    "!gsutil cp data/rsem/*.genes.results gs://hon_350/example_truncated_results/"
   ]
  }
 ],
 "metadata": {
  "environment": {
   "kernel": "python3",
   "name": "r-cpu.4-2.m102",
   "type": "gcloud",
   "uri": "gcr.io/deeplearning-platform-release/r-cpu.4-2:m102"
  },
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
